// react-tailblocks/dist/index.js
function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = require('react');
var React__default = _interopDefault(React);
var outline = require('@heroicons/react/outline');
var reactTable = require('react-table');
var ImageGallery = _interopDefault(require('react-image-gallery'));
var reactHookForm = require('react-hook-form');

function Blog1(_ref) {
  var title = _ref.title,
      items = _ref.items,
      className = _ref.className,
      cardClass = _ref.cardClass;

  var _title = title || /*#__PURE__*/React__default.createElement("h1", {
    className: "pt-24 sm:text-3xl text-2xl font-medium title-font text-center text-gray-900 mb-20"
  }, "Effervescent Blog Articles");

  var _cardClass = cardClass || "md:w-1/3";

  var _items = items || [{
    title: "The Catalyzer",
    category: "Category",
    description: "Photo booth fam kinfolk cold-pressed sriracha leggings jianbing microdosing tousled waistcoat.",
    image: "https://dummyimage.com/720x400",
    more: "Learn More",
    views: "1200",
    comments: "6"
  }, {
    title: "The 400 Blows",
    category: "Category",
    description: "Photo booth fam kinfolk cold-pressed sriracha leggings jianbing microdosing tousled waistcoat. Photo booth fam kinfolk cold-pressed sriracha leggings jianbing microdosing tousled waistcoat. Photo booth fam kinfolk cold-pressed sriracha leggings jianbing microdosing tousled waistcoat.",
    image: "https://dummyimage.com/720x400",
    more: "Learn More",
    views: "1200",
    comments: "6"
  }, {
    title: "Shooting Stars",
    category: "Category",
    description: "Photo booth fam kinfolk cold-pressed sriracha leggings jianbing microdosing tousled waistcoat.",
    image: "https://dummyimage.com/720x400",
    more: "Learn More",
    views: "1.2K",
    comments: "6"
  }];

  return /*#__PURE__*/React__default.createElement("section", {
    className: "text-gray-600 body-font " + (className ? className : "")
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "container px-5 py-24 mx-auto"
  }, typeof _title === "string" ? /*#__PURE__*/React__default.createElement("h1", {
    className: "sm:text-3xl text-2xl font-medium title-font text-center text-gray-900 mb-20"
  }, _title) : _title, /*#__PURE__*/React__default.createElement("div", {
    className: "flex flex-wrap -m-4"
  }, _items.map(function (d, i) {
    return /*#__PURE__*/React__default.createElement("div", {
      className: "p-4 " + _cardClass,
      key: i
    }, /*#__PURE__*/React__default.createElement("div", {
      className: "h-full border-2 border-gray-200 border-opacity-60 rounded-lg overflow-hidden"
    }, /*#__PURE__*/React__default.createElement("div", {
      className: "flex flex-col h-full"
    }, typeof d.image === "string" ? /*#__PURE__*/React__default.createElement("img", {
      className: "lg:h-48 md:h-36 w-full object-cover object-center",
      src: d.image,
      alt: "blog"
    }) : d.image, typeof d.category === "string" ? /*#__PURE__*/React__default.createElement("h2", {
      className: "px-6 pt-4 uppercase tracking-widest text-xs title-font font-medium text-gray-400 mb-1"
    }, d.category) : d.category, typeof d.title === "string" ? /*#__PURE__*/React__default.createElement("h1", {
      className: "px-6 title-font text-lg font-medium text-gray-900 mb-3"
    }, d.title) : d.title, typeof d.description === "string" ? /*#__PURE__*/React__default.createElement("p", {
      className: "px-6 leading-relaxed mb-3 flex-grow"
    }, d.description) : d.description, /*#__PURE__*/React__default.createElement("div", {
      className: "flex items-center flex-wrap px-6 pb-4"
    }, typeof d.more === "string" ? /*#__PURE__*/React__default.createElement("a", {
      className: "text-blue-500 inline-flex items-center md:mb-2 lg:mb-0"
    }, d.more, /*#__PURE__*/React__default.createElement(outline.ArrowSmRightIcon, {
      className: "ml-4 h-5 w-5"
    })) : d.more, typeof d.views === "string" ? /*#__PURE__*/React__default.createElement("span", {
      className: "text-gray-400 mr-3 inline-flex items-center lg:ml-auto md:ml-0 ml-auto leading-none text-sm py-1"
    }, /*#__PURE__*/React__default.createElement(outline.EyeIcon, {
      className: "h-5 w-5"
    }), d.views) : d.views, typeof d.comments === "string" ? /*#__PURE__*/React__default.createElement("span", {
      className: "text-gray-400 inline-flex items-center leading-none text-sm pl-3 border-l-2 border-gray-200"
    }, /*#__PURE__*/React__default.createElement(outline.ChatIcon, {
      className: "h-5 w-5"
    }), d.comments) : d.comments))));
  }))));
}

function Card1(_ref) {
  var title = _ref.title,
      value = _ref.value,
      icon = _ref.icon,
      className = _ref.className;

  var _title = title || "Inventory";

  var _value = value || "23,282";

  var _icon = icon || /*#__PURE__*/React__default.createElement("div", {
    className: "p-3 rounded-full bg-indigo-600 bg-opacity-75"
  }, /*#__PURE__*/React__default.createElement(outline.ShoppingBagIcon, {
    className: "h-8 w-8"
  }));

  return /*#__PURE__*/React__default.createElement("div", {
    className: "flex items-center px-5 py-6 shadow-sm rounded-md bg-white " + (className ? className : "")
  }, _icon, /*#__PURE__*/React__default.createElement("div", {
    className: "mx-5"
  }, typeof _value === "string" ? /*#__PURE__*/React__default.createElement("h4", {
    className: "text-2xl font-semibold text-gray-700"
  }, _value) : _value, typeof _title === "string" ? /*#__PURE__*/React__default.createElement("div", {
    className: "text-gray-500"
  }, _title) : _title));
}

function Content3(_ref) {
  var title = _ref.title,
      description = _ref.description,
      items = _ref.items,
      button = _ref.button,
      className = _ref.className;

  var _title = title || "Pitchfork Kickstarter Taxidermy";

  var _description = description || "Whatever cardigan tote bag tumblr hexagon brooklyn asymmetrical gentrify, subway tile poke farm-to-table.";

  var _button = /*#__PURE__*/React__default.createElement("button", {
    className: "flex mx-auto mt-16 text-white bg-blue-500 border-0 py-2 px-8 focus:outline-none hover:bg-blue-600 rounded text-lg"
  }, "Button");

  if (button !== undefined) {
    if (button && typeof button.title === "string") {
      _button = /*#__PURE__*/React__default.createElement("button", {
        className: "flex mx-auto mt-16 text-white bg-blue-500 border-0 py-2 px-8 focus:outline-none hover:bg-blue-600 rounded text-lg",
        onClick: button.onClick ? button.onClick : function () {}
      }, button.title);
    } else {
      _button = button;
    }
  }

  var _items = items || [{
    title: "Shooting Stars",
    description: "Fingerstache flexitarian street art 8-bit waist co, subway tile poke farm.",
    icon: /*#__PURE__*/React__default.createElement(outline.SparklesIcon, {
      className: "w-6 h-6"
    })
  }, {
    title: "The Catalyzer",
    description: "Fingerstache flexitarian street art 8-bit waist co, subway tile poke farm.",
    icon: /*#__PURE__*/React__default.createElement(outline.BeakerIcon, {
      className: "w-6 h-6"
    })
  }, {
    title: "Neptune",
    description: "Fingerstache flexitarian street art 8-bit waist co, subway tile poke farm.",
    icon: /*#__PURE__*/React__default.createElement(outline.GlobeIcon, {
      className: "w-6 h-6"
    })
  }, {
    title: "Melanchole",
    description: "Fingerstache flexitarian street art 8-bit waist co, subway tile poke farm.",
    icon: /*#__PURE__*/React__default.createElement(outline.CalculatorIcon, {
      className: "w-6 h-6"
    })
  }, {
    title: "Bunker",
    description: "Fingerstache flexitarian street art 8-bit waist co, subway tile poke farm.",
    icon: /*#__PURE__*/React__default.createElement(outline.LockClosedIcon, {
      className: "w-6 h-6"
    })
  }, {
    title: "Ramona Falls",
    description: "Fingerstache flexitarian street art 8-bit waist co, subway tile poke farm.",
    icon: /*#__PURE__*/React__default.createElement(outline.ColorSwatchIcon, {
      className: "w-6 h-6"
    })
  }];

  return /*#__PURE__*/React__default.createElement("section", {
    className: "text-gray-600 body-font " + (className ? className : "")
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "container px-5 py-24 mx-auto"
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "flex flex-wrap w-full mb-20 flex-col items-center text-center"
  }, typeof _title === "string" ? /*#__PURE__*/React__default.createElement("h1", {
    className: "sm:text-3xl text-2xl font-medium title-font mb-2 text-gray-900"
  }, _title) : _title, typeof _description === "string" ? /*#__PURE__*/React__default.createElement("p", {
    className: "lg:w-1/2 w-full leading-relaxed text-gray-500"
  }, _description) : _description), /*#__PURE__*/React__default.createElement("div", {
    className: "flex flex-wrap -m-4"
  }, _items.map(function (i, c) {
    return /*#__PURE__*/React__default.createElement("div", {
      className: "xl:w-1/3 md:w-1/2 p-4",
      key: c
    }, /*#__PURE__*/React__default.createElement("div", {
      className: "border border-gray-200 p-6 rounded-lg"
    }, /*#__PURE__*/React__default.createElement("div", {
      className: "w-10 h-10 inline-flex items-center justify-center rounded-full bg-blue-100 text-blue-500 mb-4"
    }, i.icon), typeof i.title === "string" ? /*#__PURE__*/React__default.createElement("h2", {
      className: "text-lg text-gray-900 font-medium title-font mb-2"
    }, i.title) : i.title, typeof i.description === "string" ? /*#__PURE__*/React__default.createElement("p", {
      className: "leading-relaxed text-base"
    }, i.description) : i.description));
  })), _button));
}

function _extends() {
  _extends = Object.assign || function (target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];

      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }

    return target;
  };

  return _extends.apply(this, arguments);
}

function Table1(_ref) {
  var data = _ref.data,
      columns = _ref.columns,
      className = _ref.className;

  var _data = data || React.useMemo(function () {
    return [{
      name: /*#__PURE__*/React__default.createElement("div", {
        className: "flex items-center"
      }, /*#__PURE__*/React__default.createElement("div", {
        className: "flex-shrink-0 h-10 w-10"
      }, /*#__PURE__*/React__default.createElement("img", {
        className: "h-10 w-10 rounded-full",
        src: "https://images.unsplash.com/photo-1472099645785-5658abf4ff4e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80",
        alt: ""
      })), /*#__PURE__*/React__default.createElement("div", {
        className: "ml-4"
      }, /*#__PURE__*/React__default.createElement("div", {
        className: "text-sm leading-5 font-medium text-gray-900"
      }, "John Doe"), /*#__PURE__*/React__default.createElement("div", {
        className: "text-sm leading-5 text-gray-500"
      }, "john@example.com"))),
      title: /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement("div", {
        className: "text-sm leading-5 text-gray-900"
      }, "Software Engineer"), /*#__PURE__*/React__default.createElement("div", {
        className: "text-sm leading-5 text-gray-500"
      }, "Web dev")),
      status: /*#__PURE__*/React__default.createElement("span", {
        className: "px-2 inline-flex text-xs leading-5 font-semibold rounded-full bg-green-100 text-green-800"
      }, "Active"),
      role: /*#__PURE__*/React__default.createElement("span", {
        className: "text-sm leading-5 text-gray-500"
      }, "Owner"),
      edit: /*#__PURE__*/React__default.createElement("a", {
        href: "#",
        className: "text-blue-600 hover:text-blue-900 text-sm leading-5 font-medium"
      }, "Edit")
    }];
  }, []);

  var _columns = columns || React.useMemo(function () {
    return [{
      Header: 'Name',
      accessor: 'name'
    }, {
      Header: 'Title',
      accessor: 'title'
    }, {
      Header: 'Status',
      accessor: 'status'
    }, {
      Header: 'Role',
      accessor: 'role'
    }, {
      Header: '',
      accessor: 'edit'
    }];
  }, []);

  var table = reactTable.useTable({
    columns: _columns,
    data: _data
  });
  var getTableProps = table.getTableProps,
      getTableBodyProps = table.getTableBodyProps,
      headerGroups = table.headerGroups,
      rows = table.rows,
      prepareRow = table.prepareRow;
  return /*#__PURE__*/React__default.createElement("div", {
    className: "flex flex-col mt-4 " + (className ? className : "")
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "-my-2 py-2 overflow-x-auto sm:-mx-6 sm:px-6 lg:-mx-8 lg:px-8"
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "align-middle inline-block min-w-full shadow overflow-hidden sm:rounded-lg border-b border-gray-200"
  }, /*#__PURE__*/React__default.createElement("table", _extends({
    className: "min-w-full"
  }, getTableProps()), /*#__PURE__*/React__default.createElement("thead", null, headerGroups.map(function (hg) {
    return /*#__PURE__*/React__default.createElement("tr", hg.getHeaderGroupProps(), hg.headers.map(function (column) {
      return /*#__PURE__*/React__default.createElement("th", _extends({
        className: "px-6 py-3 border-b border-gray-200 bg-gray-50 text-left text-xs leading-4 font-medium text-gray-500 uppercase tracking-wider"
      }, column.getHeaderProps()), column.render('Header'));
    }));
  })), /*#__PURE__*/React__default.createElement("tbody", _extends({
    className: "bg-white"
  }, getTableBodyProps()), rows.map(function (r) {
    prepareRow(r);
    return /*#__PURE__*/React__default.createElement("tr", r.getRowProps(), r.cells.map(function (c) {
      return /*#__PURE__*/React__default.createElement("td", _extends({
        className: "px-6 py-4 whitespace-no-wrap border-b border-gray-200"
      }, c.getCellProps()), c.render('Cell'));
    }));
  }))))));
}

function Logo1(_ref) {
  var title = _ref.title,
      image = _ref.image,
      inverse = _ref.inverse,
      className = _ref.className,
      href = _ref.href;
  var txtClass = inverse ? "text-white " : "text-gray-900 ";
  var svgClass = inverse ? "text-blue-500 bg-white " : "text-white bg-blue-500 ";

  var _title = title || "React Tailblocks";

  var _svg = image || /*#__PURE__*/React__default.createElement(outline.CubeIcon, {
    className: "w-10 h-10 rounded-full p-2 " + svgClass
  });

  var _href = href || "/";

  return /*#__PURE__*/React__default.createElement("a", {
    href: _href,
    className: "flex order-first title-font font-medium items-center " + txtClass + (className ? className : "")
  }, _svg, /*#__PURE__*/React__default.createElement("span", {
    className: "ml-3 text-xl"
  }, _title));
}

function Header2(_ref) {
  var button = _ref.button,
      logo = _ref.logo,
      menu = _ref.menu,
      className = _ref.className,
      fixed = _ref.fixed,
      inverse = _ref.inverse;

  var _useState = React.useState(true),
      show = _useState[0],
      setShow = _useState[1];

  var btnStyle = inverse ? "bg-gray-100 text-gray-600 hover:bg-gray-200" : "bg-blue-500 text-white hover:bg-blue-200";

  var _button = /*#__PURE__*/React__default.createElement("button", {
    className: "flex justify-between items-center border-0 py-1 px-3 focus:outline-none rounded text-base mt-4 md:mt-0 " + btnStyle
  }, "Button", /*#__PURE__*/React__default.createElement(outline.ArrowSmRightIcon, {
    className: "ml-4 h-5 w-5"
  }));

  if (button !== undefined) {
    if (button && typeof button.title === "string") {
      _button = /*#__PURE__*/React__default.createElement("button", {
        className: "flex justify-between items-center border-0 py-1 px-3 focus:outline-none rounded text-base mt-4 md:mt-0 " + btnStyle,
        onClick: button.onClick ? button.onClick : function () {}
      }, button.title, /*#__PURE__*/React__default.createElement(outline.ArrowSmRightIcon, {
        className: "ml-4 h-5 w-5"
      }));
    } else {
      _button = button;
    }
  }

  var _logo = logo || /*#__PURE__*/React__default.createElement(Logo1, {
    inverse: inverse
  });

  var _menu = menu === undefined ? [/*#__PURE__*/React__default.createElement("a", {
    className: "md:mr-5 mb-2 md:mb-0 hover:text-gray-900 text-gray-500",
    href: "https://google.com",
    key: "1"
  }, "First Link"), /*#__PURE__*/React__default.createElement("a", {
    className: "md:mr-5 mb-2 md:mb-0 hover:text-gray-900 text-gray-500",
    href: "https://google.com",
    key: "2"
  }, "Second Link"), /*#__PURE__*/React__default.createElement("a", {
    className: "md:mr-5 mb-2 md:mb-0 hover:text-gray-900 text-gray-500",
    href: "https://google.com",
    key: "3"
  }, "Third Link"), /*#__PURE__*/React__default.createElement("a", {
    className: "md:mr-5 mb-2 md:mb-0 hover:text-gray-900 text-gray-500",
    href: "https://google.com",
    key: "4"
  }, "Fourth Link")] : menu;

  var borderStyle = _menu && _menu.length ? "md:border-l md:border-white-400" : "";
  var fixedStyle = fixed ? "md:fixed " : "";
  var inverseStyle = inverse ? "text-white bg-blue-500 " : "text-blue-500 bg-white ";
  React.useEffect(function () {
    setShow(window === undefined ? true : window.innerWidth >= 768);
  }, []);
  return /*#__PURE__*/React__default.createElement("header", {
    className: "body-font shadow-md z-50 w-full" + fixedStyle + inverseStyle + (className ? className : "")
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "p-3 md:p-5 flex flex-col md:flex-row"
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "flex justify-between md:justify-start"
  }, _logo, /*#__PURE__*/React__default.createElement(outline.MenuIcon, {
    className: "h-4 w-4 md:hidden mt-3 " + inverseStyle,
    onClick: function onClick() {
      setShow(!show);
    }
  })), show && /*#__PURE__*/React__default.createElement(React__default.Fragment, null, /*#__PURE__*/React__default.createElement("nav", {
    className: "md:mr-auto md:ml-4 md:py-2 md:pl-4 flex flex-col md:flex-row items-center mt-4 md:mt-0 " + borderStyle
  }, _menu && _menu.map(function (m) {
    return m;
  })), _button)));
}

function Dashboard1(_ref) {
  var navigation = _ref.navigation,
      menu = _ref.menu,
      contentClass = _ref.contentClass,
      selectedItem = _ref.selectedItem,
      onSelect = _ref.onSelect;

  var _useState = React.useState(true),
      show = _useState[0],
      setShow = _useState[1];

  var _contentClass = contentClass || "bg-gray-100 px-6 py-8";

  var _navigation = navigation || /*#__PURE__*/React__default.createElement(Header2, null);

  var _menu = menu || [{
    category: "Manage",
    items: [{
      title: "Dashboard",
      icon: /*#__PURE__*/React__default.createElement(outline.ChartPieIcon, {
        className: "h-6 w-6"
      }),
      content: /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement("h3", {
        className: "text-gray-700 text-3xl font-medium"
      }, "Dashboard"), /*#__PURE__*/React__default.createElement("div", {
        className: "mt-8"
      }), /*#__PURE__*/React__default.createElement("div", {
        className: "flex flex-wrap -mx-6"
      }, /*#__PURE__*/React__default.createElement("div", {
        className: "w-full px-6 sm:w-1/2 xl:w-1/3"
      }, /*#__PURE__*/React__default.createElement(Card1, {
        title: "New Users",
        value: "8,282",
        icon: /*#__PURE__*/React__default.createElement("div", {
          className: "p-3 rounded-full bg-green-600 bg-opacity-75 text-white"
        }, /*#__PURE__*/React__default.createElement(outline.UserGroupIcon, {
          className: "h-8 w-8"
        }))
      })), /*#__PURE__*/React__default.createElement("div", {
        className: "w-full mt-6 px-6 sm:w-1/2 xl:w-1/3 sm:mt-0"
      }, /*#__PURE__*/React__default.createElement(Card1, {
        title: "Total Orders",
        value: "200,532",
        icon: /*#__PURE__*/React__default.createElement("div", {
          className: "p-3 rounded-full bg-yellow-600 bg-opacity-75 text-white"
        }, /*#__PURE__*/React__default.createElement(outline.CollectionIcon, {
          className: "h-8 w-8"
        }))
      })), /*#__PURE__*/React__default.createElement("div", {
        className: "w-full mt-6 px-6 sm:w-1/2 xl:w-1/3 xl:mt-0"
      }, /*#__PURE__*/React__default.createElement(Card1, {
        title: "Available Products",
        value: "215,548",
        icon: /*#__PURE__*/React__default.createElement("div", {
          className: "p-3 rounded-full bg-indigo-600 bg-opacity-75 text-white"
        }, /*#__PURE__*/React__default.createElement(outline.ShoppingBagIcon, {
          className: "h-8 w-8"
        }))
      }))))
    }, {
      title: "Users",
      icon: /*#__PURE__*/React__default.createElement(outline.UserGroupIcon, {
        className: "h-6 w-6"
      }),
      content: /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement("h3", {
        className: "text-gray-700 text-3xl font-medium"
      }, "Users"), /*#__PURE__*/React__default.createElement("div", {
        className: "mt-8"
      }), /*#__PURE__*/React__default.createElement(Table1, null))
    }, {
      title: "Applications",
      icon: /*#__PURE__*/React__default.createElement(outline.CollectionIcon, {
        className: "h-6 w-6"
      }),
      content: /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement("h3", {
        className: "text-gray-700 text-3xl font-medium"
      }, "Applications"))
    }]
  }, {
    category: "Configure",
    items: [{
      title: "Elements",
      icon: /*#__PURE__*/React__default.createElement(outline.CubeIcon, {
        className: "h-6 w-6"
      }),
      content: /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement("h3", {
        className: "text-gray-700 text-3xl font-medium"
      }, "Elements"))
    }, {
      title: "Tables",
      icon: /*#__PURE__*/React__default.createElement(outline.TableIcon, {
        className: "h-6 w-6"
      }),
      content: /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement("h3", {
        className: "text-gray-700 text-3xl font-medium"
      }, "Tables"))
    }, {
      title: "Forms",
      icon: /*#__PURE__*/React__default.createElement(outline.TemplateIcon, {
        className: "h-6 w-6"
      }),
      content: /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement("h3", {
        className: "text-gray-700 text-3xl font-medium"
      }, "Forms"))
    }]
  }, {
    items: [{
      content: /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement("h3", {
        className: "text-gray-700 text-3xl font-medium"
      }, "Profile"))
    }]
  }];

  var _useState2 = React.useState(),
      selected = _useState2[0],
      setSelected = _useState2[1];

  React.useEffect(function () {
    setSelected(selectedItem ? selectedItem : _menu[0].items[0]);
    if (onSelect) onSelect(selectedItem ? selectedItem : _menu[0].items[0]);
    setShow(window === undefined ? true : window.innerWidth >= 768);
  }, [selectedItem]);
  return /*#__PURE__*/React__default.createElement("div", {
    className: "flex flex-col h-screen"
  }, _navigation, /*#__PURE__*/React__default.createElement("div", {
    className: "flex h-full bg-gray-200"
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "z-50 inset-y-0 left-0 transition h-full duration-300 transform bg-gray-900 overflow-hidden lg:translate-x-0 lg:static lg:inset-0"
  }, /*#__PURE__*/React__default.createElement("nav", {
    className: "md:mt-8 text-gray-500 h-full flex " + (show ? "mt-4 md:mt-8" : "")
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "w-full overflow-x-hidden overflow-y-auto"
  }, _menu.map(function (m, i) {
    return m.category && /*#__PURE__*/React__default.createElement("div", {
      key: i,
      className: "pl-2 md:pl-4 pb-8"
    }, show ? /*#__PURE__*/React__default.createElement("h2", {
      className: "text-lg pb-2"
    }, m.category) : /*#__PURE__*/React__default.createElement("div", {
      className: "mt-2"
    }), m.items.map(function (t, j) {
      var hover = (selected === null || selected === void 0 ? void 0 : selected.title) === t.title ? "bg-gray-500 bg-opacity-25 text-gray-300" : "hover:bg-gray-700 hover:bg-opacity-25 hover:text-gray-100";
      return /*#__PURE__*/React__default.createElement("div", {
        key: j,
        className: "flex items-center py-1 md:py-2 px-1 md:px-4 cursor-pointer " + hover,
        onClick: function onClick(e) {
          setSelected(t);
          if (onSelect) onSelect(t);
        }
      }, t.icon, show && /*#__PURE__*/React__default.createElement("span", {
        className: "mx-3"
      }, t.title));
    }));
  })), /*#__PURE__*/React__default.createElement("div", {
    className: "m-auto overflow-hidden"
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "text-3xl md:text-4xl pr-1 cursor-pointer",
    onClick: function onClick() {
      setShow(!show);
    }
  }, "|")))), /*#__PURE__*/React__default.createElement("div", {
    className: "flex-1 flex flex-column overflow-hidden"
  }, /*#__PURE__*/React__default.createElement("main", {
    className: "flex-1 overflow-x-hidden overflow-y-auto " + _contentClass
  }, selected === null || selected === void 0 ? void 0 : selected.content))));
}

function Ecommerce1(_ref) {
  var items = _ref.items,
      className = _ref.className;

  var _items = items || [{
    image: /*#__PURE__*/React__default.createElement("a", {
      className: "block relative h-48 rounded overflow-hidden"
    }, /*#__PURE__*/React__default.createElement("img", {
      alt: "ecommerce",
      className: "object-cover object-center w-full h-full block",
      src: "https://dummyimage.com/420x260"
    })),
    category: /*#__PURE__*/React__default.createElement("h3", {
      className: "text-gray-500 text-xs tracking-widest title-font mb-1"
    }, "CATEGORY"),
    title: /*#__PURE__*/React__default.createElement("h2", {
      className: "text-gray-900 title-font text-lg font-medium"
    }, "The Catalyzer"),
    description: /*#__PURE__*/React__default.createElement("p", {
      className: "mt-1 text-sm"
    }, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer nec dui magna. Nam vehicula risus at pulvinar blandit. Donec eros urna, cursus id rutrum in, accumsan aliquam est."),
    price: /*#__PURE__*/React__default.createElement("p", {
      className: "mt-2"
    }, "$16.00")
  }, {
    image: /*#__PURE__*/React__default.createElement("a", {
      className: "block relative h-48 rounded overflow-hidden"
    }, /*#__PURE__*/React__default.createElement("img", {
      alt: "ecommerce",
      className: "object-cover object-center w-full h-full block",
      src: "https://dummyimage.com/420x260"
    })),
    category: /*#__PURE__*/React__default.createElement("h3", {
      className: "text-gray-500 text-xs tracking-widest title-font mb-1"
    }, "CATEGORY"),
    title: /*#__PURE__*/React__default.createElement("h2", {
      className: "text-gray-900 title-font text-lg font-medium"
    }, "Shooting Stars"),
    description: /*#__PURE__*/React__default.createElement("p", {
      className: "mt-1 text-sm"
    }, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer nec dui magna. Nam vehicula risus at pulvinar blandit. Donec eros urna, cursus id rutrum in, accumsan aliquam est."),
    price: /*#__PURE__*/React__default.createElement("p", {
      className: "mt-2"
    }, "$16.00")
  }, {
    image: /*#__PURE__*/React__default.createElement("a", {
      className: "block relative h-48 rounded overflow-hidden"
    }, /*#__PURE__*/React__default.createElement("img", {
      alt: "ecommerce",
      className: "object-cover object-center w-full h-full block",
      src: "https://dummyimage.com/420x260"
    })),
    category: /*#__PURE__*/React__default.createElement("h3", {
      className: "text-gray-500 text-xs tracking-widest title-font mb-1"
    }, "CATEGORY"),
    title: /*#__PURE__*/React__default.createElement("h2", {
      className: "text-gray-900 title-font text-lg font-medium"
    }, "Neptune"),
    description: /*#__PURE__*/React__default.createElement("p", {
      className: "mt-1 text-sm"
    }, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer nec dui magna. Nam vehicula risus at pulvinar blandit. Donec eros urna, cursus id rutrum in, accumsan aliquam est."),
    price: /*#__PURE__*/React__default.createElement("p", {
      className: "mt-2"
    }, "$16.00")
  }, {
    image: /*#__PURE__*/React__default.createElement("a", {
      className: "block relative h-48 rounded overflow-hidden"
    }, /*#__PURE__*/React__default.createElement("img", {
      alt: "ecommerce",
      className: "object-cover object-center w-full h-full block",
      src: "https://dummyimage.com/420x260"
    })),
    category: /*#__PURE__*/React__default.createElement("h3", {
      className: "text-gray-500 text-xs tracking-widest title-font mb-1"
    }, "CATEGORY"),
    title: /*#__PURE__*/React__default.createElement("h2", {
      className: "text-gray-900 title-font text-lg font-medium"
    }, "The 400 Blows"),
    description: /*#__PURE__*/React__default.createElement("p", {
      className: "mt-1 text-sm"
    }, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer nec dui magna. Nam vehicula risus at pulvinar blandit. Donec eros urna, cursus id rutrum in, accumsan aliquam est."),
    price: /*#__PURE__*/React__default.createElement("p", {
      className: "mt-2"
    }, "$16.00")
  }, {
    image: /*#__PURE__*/React__default.createElement("a", {
      className: "block relative h-48 rounded overflow-hidden"
    }, /*#__PURE__*/React__default.createElement("img", {
      alt: "ecommerce",
      className: "object-cover object-center w-full h-full block",
      src: "https://dummyimage.com/420x260"
    })),
    category: /*#__PURE__*/React__default.createElement("h3", {
      className: "text-gray-500 text-xs tracking-widest title-font mb-1"
    }, "CATEGORY"),
    title: /*#__PURE__*/React__default.createElement("h2", {
      className: "text-gray-900 title-font text-lg font-medium"
    }, "The Catalyzer"),
    description: /*#__PURE__*/React__default.createElement("p", {
      className: "mt-1 text-sm"
    }, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer nec dui magna. Nam vehicula risus at pulvinar blandit. Donec eros urna, cursus id rutrum in, accumsan aliquam est."),
    price: /*#__PURE__*/React__default.createElement("p", {
      className: "mt-2"
    }, "$16.00")
  }, {
    image: /*#__PURE__*/React__default.createElement("a", {
      className: "block relative h-48 rounded overflow-hidden"
    }, /*#__PURE__*/React__default.createElement("img", {
      alt: "ecommerce",
      className: "object-cover object-center w-full h-full block",
      src: "https://dummyimage.com/420x260"
    })),
    category: /*#__PURE__*/React__default.createElement("h3", {
      className: "text-gray-500 text-xs tracking-widest title-font mb-1"
    }, "CATEGORY"),
    title: /*#__PURE__*/React__default.createElement("h2", {
      className: "text-gray-900 title-font text-lg font-medium"
    }, "Shooting Stars"),
    description: /*#__PURE__*/React__default.createElement("p", {
      className: "mt-1 text-sm"
    }, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer nec dui magna. Nam vehicula risus at pulvinar blandit. Donec eros urna, cursus id rutrum in, accumsan aliquam est."),
    price: /*#__PURE__*/React__default.createElement("p", {
      className: "mt-2"
    }, "$16.00")
  }, {
    image: /*#__PURE__*/React__default.createElement("a", {
      className: "block relative h-48 rounded overflow-hidden"
    }, /*#__PURE__*/React__default.createElement("img", {
      alt: "ecommerce",
      className: "object-cover object-center w-full h-full block",
      src: "https://dummyimage.com/420x260"
    })),
    category: /*#__PURE__*/React__default.createElement("h3", {
      className: "text-gray-500 text-xs tracking-widest title-font mb-1"
    }, "CATEGORY"),
    title: /*#__PURE__*/React__default.createElement("h2", {
      className: "text-gray-900 title-font text-lg font-medium"
    }, "Neptune"),
    description: /*#__PURE__*/React__default.createElement("p", {
      className: "mt-1 text-sm"
    }, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer nec dui magna. Nam vehicula risus at pulvinar blandit. Donec eros urna, cursus id rutrum in, accumsan aliquam est."),
    price: /*#__PURE__*/React__default.createElement("p", {
      className: "mt-2"
    }, "$16.00")
  }, {
    image: /*#__PURE__*/React__default.createElement("a", {
      className: "block relative h-48 rounded overflow-hidden"
    }, /*#__PURE__*/React__default.createElement("img", {
      alt: "ecommerce",
      className: "object-cover object-center w-full h-full block",
      src: "https://dummyimage.com/420x260"
    })),
    category: /*#__PURE__*/React__default.createElement("h3", {
      className: "text-gray-500 text-xs tracking-widest title-font mb-1"
    }, "CATEGORY"),
    title: /*#__PURE__*/React__default.createElement("h2", {
      className: "text-gray-900 title-font text-lg font-medium"
    }, "The 400 Blows"),
    description: /*#__PURE__*/React__default.createElement("p", {
      className: "mt-1 text-sm"
    }, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer nec dui magna. Nam vehicula risus at pulvinar blandit. Donec eros urna, cursus id rutrum in, accumsan aliquam est."),
    price: /*#__PURE__*/React__default.createElement("p", {
      className: "mt-2"
    }, "$16.00")
  }];

  return /*#__PURE__*/React__default.createElement("section", {
    className: "text-gray-600 body-font " + (className ? className : "")
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "container px-5 py-24 mx-auto"
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "flex flex-wrap -m-4"
  }, _items.map(function (i, c) {
    return /*#__PURE__*/React__default.createElement("div", {
      className: "lg:w-1/4 md:w-1/2 p-4 w-full",
      key: c
    }, typeof i.image === "string" ? /*#__PURE__*/React__default.createElement("a", {
      className: "block relative h-48 rounded overflow-hidden"
    }, /*#__PURE__*/React__default.createElement("img", {
      alt: "ecommerce",
      className: "object-cover object-center w-full h-full block",
      src: i.image
    })) : i.image, /*#__PURE__*/React__default.createElement("div", {
      className: "mt-4"
    }, typeof i.category === "string" ? /*#__PURE__*/React__default.createElement("h3", {
      className: "text-gray-500 text-xs tracking-widest title-font mb-1"
    }, i.category) : i.category, typeof i.title === "string" ? /*#__PURE__*/React__default.createElement("h2", {
      className: "text-gray-900 title-font text-lg font-medium"
    }, i.title) : i.title, typeof i.description === "string" ? /*#__PURE__*/React__default.createElement("p", {
      className: "mt-1 text-sm"
    }, i.description) : i.description, typeof i.price === "string" || typeof i.price === "number" ? /*#__PURE__*/React__default.createElement("p", {
      className: "mt-2"
    }, "$", i.price) : i.price));
  }))));
}

function Ecommerce2(_ref) {
  var brand = _ref.brand,
      title = _ref.title,
      buttonText = _ref.buttonText,
      description = _ref.description,
      variant = _ref.variant,
      images = _ref.images,
      reviews = _ref.reviews,
      className = _ref.className,
      color = _ref.color,
      onPurchase = _ref.onPurchase;

  var _txtcolor = color ? {
    color: color
  } : {};

  var _bgcolor = color ? {
    backgroundColor: color
  } : {};

  var _brand = brand || /*#__PURE__*/React__default.createElement("h2", {
    className: "text-sm title-font text-gray-500 tracking-widest"
  }, "BRAND NAME");

  var _title = title || /*#__PURE__*/React__default.createElement("h1", {
    className: "text-gray-900 text-3xl title-font font-medium mb-1"
  }, "The Catcher in the Rye");

  var _buttonText = buttonText || "Purchase";

  var _description = description || /*#__PURE__*/React__default.createElement("p", {
    className: "leading-relaxed"
  }, "Fam locavore kickstarter distillery.Mixtape chillwave tumeric sriracha taximy chia microdosing tilde DIY.XOXO fam indxgo juiceramps cornhole raw denim forage brooklyn.Everyday carry +1 seitan poutine tumeric.Gastropub blue bottle austin listicle pour-over, neutra jean shorts keytar banjo tattooed umami cardigan.");

  var _variant = variant || {
    title: "Size",
    options: [{
      title: "SM",
      price: 58.01
    }, {
      title: "M",
      price: 58.02
    }, {
      title: "L",
      price: 58.03
    }, {
      title: "XL",
      price: 58.04
    }]
  };

  if (!_variant.options || !_variant.options.length) _variant.options = [{
    price: 0.00
  }];

  var _images = images !== undefined ? images : [{
    original: 'https://picsum.photos/id/1018/400/800/',
    thumbnail: 'https://picsum.photos/id/1018/100/100/'
  }, {
    original: 'https://picsum.photos/id/1015/400/800/',
    thumbnail: 'https://picsum.photos/id/1015/100/100/'
  }, {
    original: 'https://picsum.photos/id/1019/400/800/',
    thumbnail: 'https://picsum.photos/id/1019/100/100/'
  }];

  var _reviews = reviews !== undefined ? reviews : [{
    title: "Awesome Socks",
    description: "I just cant beleive the quality of these socks.",
    date: "2021-10-01",
    reviewer: "Tom Brady",
    rating: 5
  }, {
    title: "What a product",
    description: "Love these socks, just wish they had toes.",
    date: "2021-09-08",
    reviewer: "Kevin Bacon",
    rating: 4
  }, {
    title: "Really Good But ...",
    description: "Really comfortable sock, but need to be more breathable.",
    date: "2021-10-01",
    reviewer: "Jennifer Jones",
    rating: 3
  }, {
    title: "Love to Roll",
    description: "Love to roll with my new fancy socks.",
    date: "2021-10-01",
    reviewer: "Lil Wayne",
    rating: 4
  }];

  var _stars = [];
  var rating = _reviews ? _reviews.reduce(function (avg, r, _, _ref2) {
    var length = _ref2.length;
    var value = r.rating <= 5 ? r.rating : 5;
    return avg + value / length;
  }, 0) : 0;

  if (_reviews) {
    for (var i = 0; i < 5; i++) {
      if (i <= Math.round(rating) - 1) {
        _stars.push( /*#__PURE__*/React__default.createElement("svg", {
          key: i,
          fill: "currentColor",
          stroke: "currentColor",
          strokeLinecap: "round",
          strokeLinejoin: "round",
          strokeWidth: "2",
          className: "w-4 h-4 text-blue-500",
          viewBox: "0 0 24 24",
          style: _txtcolor
        }, /*#__PURE__*/React__default.createElement("path", {
          d: "M12 2l3.09 6.26L22 9.27l-5 4.87 1.18 6.88L12 17.77l-6.18 3.25L7 14.14 2 9.27l6.91-1.01L12 2z"
        })));
      } else {
        _stars.push( /*#__PURE__*/React__default.createElement("svg", {
          key: i,
          fill: "none",
          stroke: "currentColor",
          strokeLinecap: "round",
          strokeLinejoin: "round",
          strokeWidth: "2",
          className: "w-4 h-4 text-blue-500",
          viewBox: "0 0 24 24",
          style: _txtcolor
        }, /*#__PURE__*/React__default.createElement("path", {
          d: "M12 2l3.09 6.26L22 9.27l-5 4.87 1.18 6.88L12 17.77l-6.18 3.25L7 14.14 2 9.27l6.91-1.01L12 2z"
        })));
      }
    }
  }

  var _useState = React.useState(_variant.options[0]),
      option = _useState[0],
      setOption = _useState[1];

  return /*#__PURE__*/React__default.createElement("section", {
    className: "text-gray-600 body-font overflow-hidden " + (className ? className : "")
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "container px-5 py-24 mx-auto"
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "lg:w-4/5 mx-auto flex flex-wrap"
  }, _images && /*#__PURE__*/React__default.createElement("div", {
    className: "lg:w-1/2 w-full lg:h-auto h-64"
  }, /*#__PURE__*/React__default.createElement(ImageGallery, {
    thumbnailPosition: "left",
    useBrowserFullscreen: true,
    showBullets: true,
    showNav: false,
    items: _images
  })), /*#__PURE__*/React__default.createElement("div", {
    className: "lg:w-1/2 w-full lg:pl-10 lg:py-6 mt-6 lg:mt-0"
  }, typeof _brand === "string" ? /*#__PURE__*/React__default.createElement("h2", {
    className: "text-sm title-font text-gray-500 tracking-widest"
  }, _brand) : _brand, typeof _title === "string" ? /*#__PURE__*/React__default.createElement("h1", {
    className: "text-gray-900 text-3xl title-font font-medium mb-1"
  }, _title) : _title, _reviews && /*#__PURE__*/React__default.createElement("div", {
    className: "flex"
  }, /*#__PURE__*/React__default.createElement("span", {
    className: "flex items-center"
  }, _stars, /*#__PURE__*/React__default.createElement("span", {
    className: "text-gray-600 ml-3 font-light text-sm"
  }, _reviews.length, " Reviews"))), /*#__PURE__*/React__default.createElement("div", {
    className: "mb-4"
  }), typeof _description === "string" ? /*#__PURE__*/React__default.createElement("p", {
    className: "leading-relaxed"
  }, _description) : _description, option.description && /*#__PURE__*/React__default.createElement("p", {
    className: "leading-relaxed"
  }, option.description), /*#__PURE__*/React__default.createElement("div", {
    className: "flex mt-4 items-center pb-5 border-b-2 border-gray-100 mb-5"
  }, _variant.options && _variant.options.length > 1 && /*#__PURE__*/React__default.createElement("div", {
    className: "flex ml-6 items-center"
  }, /*#__PURE__*/React__default.createElement("span", {
    className: "mr-3"
  }, _variant.title), /*#__PURE__*/React__default.createElement("div", {
    className: "relative"
  }, /*#__PURE__*/React__default.createElement(React__default.Fragment, null, /*#__PURE__*/React__default.createElement("select", {
    className: "rounded border appearance-none border-gray-300 py-2 focus:outline-none focus:ring-2 focus:ring-blue-200 focus:border-blue-500 text-base pl-3 pr-10",
    value: option.title,
    onChange: function onChange(e) {
      var opt = _variant.options.find(function (o) {
        return o.title === e.target.value;
      });

      setOption(opt);
    }
  }, _variant.options.map(function (v, i) {
    return typeof v.title === "string" ? /*#__PURE__*/React__default.createElement("option", {
      key: i,
      disabled: v.disabled,
      value: v.title
    }, v.title) : v;
  })), /*#__PURE__*/React__default.createElement("span", {
    className: "absolute right-0 top-0 h-full w-10 text-center text-gray-600 pointer-events-none flex items-center justify-center"
  }, /*#__PURE__*/React__default.createElement("svg", {
    fill: "none",
    stroke: "currentColor",
    strokeLinecap: "round",
    strokeLinejoin: "round",
    strokeWidth: "2",
    className: "w-4 h-4",
    viewBox: "0 0 24 24"
  }, /*#__PURE__*/React__default.createElement("path", {
    d: "M6 9l6 6 6-6"
  }))))))), /*#__PURE__*/React__default.createElement("div", {
    className: "flex"
  }, /*#__PURE__*/React__default.createElement("span", {
    className: "title-font font-medium text-2xl text-gray-900"
  }, "$", option.price ? option.price.toFixed(2) : "0.00"), /*#__PURE__*/React__default.createElement("button", {
    className: "flex ml-auto text-white bg-blue-500 border-0 py-2 px-6 focus:outline-none rounded",
    style: _bgcolor,
    onClick: function onClick() {
      if (onPurchase) onPurchase(option);
    }
  }, _buttonText))))));
}

function Email1(_ref) {
  var title = _ref.title,
      description = _ref.description,
      button = _ref.button,
      className = _ref.className;

  var _title = title || "Get the Latest News and Updates from React Tailblocks";

  var _description = description || "Poke slow-carb mixtape knausgaard, typewriter street art gentrify hammock starladder roathse. Craies vegan tousled etsy austin.";

  var _button = /*#__PURE__*/React__default.createElement("button", {
    className: "text-white bg-blue-500 border-0 py-2 px-8 focus:outline-none hover:bg-blue-600 rounded text-lg"
  }, "Button");

  if (button !== undefined) {
    if (button && typeof button.title === "string") {
      _button = /*#__PURE__*/React__default.createElement("button", {
        className: "text-white bg-blue-500 border-0 py-2 px-8 focus:outline-none hover:bg-blue-600 rounded text-lg",
        onClick: button.onClick ? button.onClick : function () {}
      }, button.title);
    } else {
      _button = button;
    }
  }

  return /*#__PURE__*/React__default.createElement("section", {
    className: "text-gray-600 body-font " + (className ? className : "")
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "container px-5 py-24 mx-auto flex flex-wrap items-center"
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "lg:w-3/5 md:w-1/2 md:pr-16 lg:pr-0 pr-0"
  }, typeof _title === "string" ? /*#__PURE__*/React__default.createElement("h1", {
    className: "title-font font-medium text-3xl text-gray-900"
  }, _title) : _title, typeof _description === "string" ? /*#__PURE__*/React__default.createElement("p", {
    className: "leading-relaxed mt-4"
  }, _description) : _description), /*#__PURE__*/React__default.createElement("div", {
    className: "lg:w-2/6 md:w-1/2 bg-gray-100 rounded-lg p-8 flex flex-col md:ml-auto w-full mt-10 md:mt-0"
  }, /*#__PURE__*/React__default.createElement("h2", {
    className: "text-gray-900 text-lg font-medium title-font mb-5"
  }, "Sign Up"), /*#__PURE__*/React__default.createElement("div", {
    className: "relative mb-4"
  }, /*#__PURE__*/React__default.createElement("label", {
    htmlFor: "full-name",
    className: "leading-7 text-sm text-gray-600"
  }, "Full Name"), /*#__PURE__*/React__default.createElement("input", {
    type: "text",
    id: "full-name",
    name: "full-name",
    className: "w-full bg-white rounded border border-gray-300 focus:border-blue-500 focus:ring-2 focus:ring-blue-200 text-base outline-none text-gray-700 py-1 px-3 leading-8 transition-colors duration-200 ease-in-out"
  })), /*#__PURE__*/React__default.createElement("div", {
    className: "relative mb-4"
  }, /*#__PURE__*/React__default.createElement("label", {
    htmlFor: "email",
    className: "leading-7 text-sm text-gray-600"
  }, "Email"), /*#__PURE__*/React__default.createElement("input", {
    type: "email",
    id: "email",
    name: "email",
    className: "w-full bg-white rounded border border-gray-300 focus:border-blue-500 focus:ring-2 focus:ring-blue-200 text-base outline-none text-gray-700 py-1 px-3 leading-8 transition-colors duration-200 ease-in-out"
  })), _button)));
}

function Feature1(_ref) {
  var title = _ref.title,
      features = _ref.features,
      className = _ref.className;

  var _title = title || /*#__PURE__*/React__default.createElement("h1", {
    className: "sm:text-3xl text-2xl font-medium title-font text-center text-gray-900 mb-20"
  }, "Raw Denim Heirloom Man Braid", /*#__PURE__*/React__default.createElement("br", {
    className: "hidden sm:block"
  }), "Selfies Wayfarers");

  var _features = features || [{
    title: "Shooting Stars",
    description: "Blue bottle crucifix vinyl post-ironic four dollar toast vegan taxidermy. Gastropub indxgo juice poutine, ramps microdosing banh mi pug VHS try-hard ugh iceland kickstarter tumblr live-edge tilde.",
    icon: /*#__PURE__*/React__default.createElement(outline.SparklesIcon, {
      className: "w-6 h-6"
    }),
    more: /*#__PURE__*/React__default.createElement("a", {
      className: "mt-3 text-blue-500 inline-flex items-center"
    }, "Learn More", /*#__PURE__*/React__default.createElement(outline.ArrowSmRightIcon, {
      className: "w-4 h-4 ml-2"
    }))
  }, {
    title: "The Catalyzer",
    description: "Blue bottle crucifix vinyl post-ironic four dollar toast vegan taxidermy. Gastropub indxgo juice poutine, ramps microdosing banh mi pug VHS try-hard ugh iceland kickstarter tumblr live-edge tilde.",
    icon: /*#__PURE__*/React__default.createElement(outline.BeakerIcon, {
      className: "w-6 h-6"
    }),
    more: /*#__PURE__*/React__default.createElement("a", {
      className: "mt-3 text-blue-500 inline-flex items-center"
    }, "Learn More", /*#__PURE__*/React__default.createElement(outline.ArrowSmRightIcon, {
      className: "w-4 h-4 ml-2"
    }))
  }, {
    title: "Neptune",
    description: "Blue bottle crucifix vinyl post-ironic four dollar toast vegan taxidermy. Gastropub indxgo juice poutine, ramps microdosing banh mi pug VHS try-hard ugh iceland kickstarter tumblr live-edge tilde.",
    icon: /*#__PURE__*/React__default.createElement(outline.GlobeIcon, {
      className: "w-6 h-6"
    }),
    more: /*#__PURE__*/React__default.createElement("a", {
      className: "mt-3 text-blue-500 inline-flex items-center"
    }, "Learn More", /*#__PURE__*/React__default.createElement(outline.ArrowSmRightIcon, {
      className: "w-4 h-4 ml-2"
    }))
  }];

  return /*#__PURE__*/React__default.createElement("section", {
    className: "text-gray-600 body-font " + (className ? className : "")
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "container px-5 py-24 mx-auto"
  }, typeof _title === "string" ? /*#__PURE__*/React__default.createElement("h1", {
    className: "sm:text-3xl text-2xl font-medium title-font text-center text-gray-900 mb-20"
  }, _title) : _title, /*#__PURE__*/React__default.createElement("div", {
    className: "flex flex-wrap sm:-m-4 -mx-4 -mb-10 -mt-4 md:space-y-0 space-y-6"
  }, _features.map(function (f, i) {
    return /*#__PURE__*/React__default.createElement("div", {
      className: "p-4 md:w-1/3 flex",
      key: i
    }, /*#__PURE__*/React__default.createElement("div", {
      className: "w-12 h-12 inline-flex items-center justify-center rounded-full bg-blue-100 text-blue-500 mb-4 flex-shrink-0"
    }, f.icon), /*#__PURE__*/React__default.createElement("div", {
      className: "flex-grow pl-6"
    }, typeof f.title === "string" ? /*#__PURE__*/React__default.createElement("h2", {
      className: "text-gray-900 text-lg title-font font-medium mb-2"
    }, f.title) : f.title, typeof _features.description === "string" ? /*#__PURE__*/React__default.createElement("p", {
      className: "leading-relaxed text-base"
    }, f.description) : f.description, /*#__PURE__*/React__default.createElement("br", null), f.more));
  }))));
}

function FacebookSvg() {
  return /*#__PURE__*/React__default.createElement("svg", {
    fill: "currentColor",
    strokeLinecap: "round",
    strokeLinejoin: "round",
    strokeWidth: "2",
    className: "w-5 h-5",
    viewBox: "0 0 24 24"
  }, /*#__PURE__*/React__default.createElement("path", {
    d: "M18 2h-3a5 5 0 00-5 5v3H7v4h3v8h4v-8h3l1-4h-4V7a1 1 0 011-1h3z"
  }));
}
function TwitterSvg() {
  return /*#__PURE__*/React__default.createElement("svg", {
    fill: "currentColor",
    strokeLinecap: "round",
    strokeLinejoin: "round",
    strokeWidth: "2",
    className: "w-5 h-5",
    viewBox: "0 0 24 24"
  }, /*#__PURE__*/React__default.createElement("path", {
    d: "M23 3a10.9 10.9 0 01-3.14 1.53 4.48 4.48 0 00-7.86 3v1A10.66 10.66 0 013 4s-4 9 5 13a11.64 11.64 0 01-7 2c9 5 20 0 20-11.5a4.5 4.5 0 00-.08-.83A7.72 7.72 0 0023 3z"
  }));
}
function InstagramSvg() {
  return /*#__PURE__*/React__default.createElement("svg", {
    fill: "none",
    stroke: "currentColor",
    strokeLinecap: "round",
    strokeLinejoin: "round",
    strokeWidth: "2",
    className: "w-5 h-5",
    viewBox: "0 0 24 24"
  }, /*#__PURE__*/React__default.createElement("rect", {
    width: "20",
    height: "20",
    x: "2",
    y: "2",
    rx: "5",
    ry: "5"
  }), /*#__PURE__*/React__default.createElement("path", {
    d: "M16 11.37A4 4 0 1112.63 8 4 4 0 0116 11.37zm1.5-4.87h.01"
  }));
}
function LinkedInSvg() {
  return /*#__PURE__*/React__default.createElement("svg", {
    fill: "currentColor",
    stroke: "currentColor",
    strokeLinecap: "round",
    strokeLinejoin: "round",
    strokeWidth: "0",
    className: "w-5 h-5",
    viewBox: "0 0 24 24"
  }, /*#__PURE__*/React__default.createElement("path", {
    stroke: "none",
    d: "M16 8a6 6 0 016 6v7h-4v-7a2 2 0 00-2-2 2 2 0 00-2 2v7h-4v-7a6 6 0 016-6zM2 9h4v12H2z"
  }), /*#__PURE__*/React__default.createElement("circle", {
    cx: "4",
    cy: "4",
    r: "2",
    stroke: "none"
  }));
}

function Footer1(_ref) {
  var title = _ref.title,
      description = _ref.description,
      svg = _ref.svg,
      copyright = _ref.copyright,
      facebook = _ref.facebook,
      twitter = _ref.twitter,
      instagram = _ref.instagram,
      linkedin = _ref.linkedin,
      menu = _ref.menu,
      className = _ref.className;

  var _title = title || "Tailblocks";

  var _description = description || "Air plant banjo lyft occupy retro adaptogen indego";

  var _svg = svg || /*#__PURE__*/React__default.createElement(outline.CubeIcon, {
    className: "w-10 h-10 rounded-full p-2 text-white bg-blue-500"
  });

  var _facebook = facebook === undefined ? "react-tailblocks" : facebook;

  var _twitter = twitter === undefined ? "react-tailblocks" : twitter;

  var _instagram = instagram === undefined ? "react-tailblocks" : instagram;

  var _linkedin = linkedin === undefined ? "react-tailblocks" : linkedin;

  var _menu = menu || [{
    category: "CATEGORY1",
    items: [/*#__PURE__*/React__default.createElement("a", {
      className: "text-gray-600 hover:text-gray-800"
    }, "First Link"), /*#__PURE__*/React__default.createElement("a", {
      className: "text-gray-600 hover:text-gray-800"
    }, "Second Link"), /*#__PURE__*/React__default.createElement("a", {
      className: "text-gray-600 hover:text-gray-800"
    }, "Third Link"), /*#__PURE__*/React__default.createElement("a", {
      className: "text-gray-600 hover:text-gray-800"
    }, "Fourth Link")]
  }, {
    category: "CATEGORY2",
    items: [/*#__PURE__*/React__default.createElement("a", {
      className: "text-gray-600 hover:text-gray-800"
    }, "First Link"), /*#__PURE__*/React__default.createElement("a", {
      className: "text-gray-600 hover:text-gray-800"
    }, "Second Link"), /*#__PURE__*/React__default.createElement("a", {
      className: "text-gray-600 hover:text-gray-800"
    }, "Third Link"), /*#__PURE__*/React__default.createElement("a", {
      className: "text-gray-600 hover:text-gray-800"
    }, "Fourth Link")]
  }, {
    category: "CATEGORY3",
    items: [/*#__PURE__*/React__default.createElement("a", {
      className: "text-gray-600 hover:text-gray-800"
    }, "First Link"), /*#__PURE__*/React__default.createElement("a", {
      className: "text-gray-600 hover:text-gray-800"
    }, "Second Link"), /*#__PURE__*/React__default.createElement("a", {
      className: "text-gray-600 hover:text-gray-800"
    }, "Third Link"), /*#__PURE__*/React__default.createElement("a", {
      className: "text-gray-600 hover:text-gray-800"
    }, "Fourth Link")]
  }, {
    category: "CATEGORY4",
    items: [/*#__PURE__*/React__default.createElement("a", {
      className: "text-gray-600 hover:text-gray-800"
    }, "First Link"), /*#__PURE__*/React__default.createElement("a", {
      className: "text-gray-600 hover:text-gray-800"
    }, "Second Link"), /*#__PURE__*/React__default.createElement("a", {
      className: "text-gray-600 hover:text-gray-800"
    }, "Third Link"), /*#__PURE__*/React__default.createElement("a", {
      className: "text-gray-600 hover:text-gray-800"
    }, "Fourth Link")]
  }];

  return /*#__PURE__*/React__default.createElement("footer", {
    className: "text-gray-600 body-font " + (className ? className : "")
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "container px-5 py-24 mx-auto flex md:items-center lg:items-start md:flex-row md:flex-nowrap flex-wrap flex-col"
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "w-64 flex-shrink-0 md:mx-0 mx-auto text-center md:text-left"
  }, /*#__PURE__*/React__default.createElement("a", {
    className: "flex title-font font-medium items-center md:justify-start justify-center text-gray-900"
  }, _svg, typeof _title === "string" ? /*#__PURE__*/React__default.createElement("span", {
    className: "ml-3 text-xl"
  }, _title) : _title), typeof _description === "string" ? /*#__PURE__*/React__default.createElement("p", {
    className: "mt-2 text-sm text-gray-500"
  }, _description) : _description), /*#__PURE__*/React__default.createElement("div", {
    className: "flex-grow flex flex-wrap md:pl-20 -mb-10 md:mt-0 mt-10 md:text-left text-center"
  }, _menu.map(function (m, c) {
    return /*#__PURE__*/React__default.createElement("div", {
      className: "lg:w-1/4 md:w-1/2 w-full px-4",
      key: c
    }, typeof m.category === "string" ? /*#__PURE__*/React__default.createElement("h2", {
      className: "title-font font-medium text-gray-900 tracking-widest text-sm mb-3"
    }, m.category) : m.category, /*#__PURE__*/React__default.createElement("nav", {
      className: "list-none mb-10"
    }, m.items.map(function (i, c) {
      return /*#__PURE__*/React__default.createElement("li", {
        key: c
      }, typeof i === "string" ? /*#__PURE__*/React__default.createElement("a", {
        className: "text-gray-600 hover:text-gray-800"
      }, "i") : i);
    })));
  }))), copyright && /*#__PURE__*/React__default.createElement("div", {
    className: "bg-gray-100"
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "container mx-auto py-4 px-5 flex flex-wrap flex-col sm:flex-row"
  }, /*#__PURE__*/React__default.createElement("p", {
    className: "text-gray-500 text-sm text-center sm:text-left"
  }, copyright, _twitter && twitter && /*#__PURE__*/React__default.createElement("span", null, " -", /*#__PURE__*/React__default.createElement("a", {
    href: "https://twitter.com/" + _twitter,
    className: "text-gray-600 ml-1",
    rel: "noopener noreferrer",
    target: "_blank"
  }, "@", _twitter))), /*#__PURE__*/React__default.createElement("span", {
    className: "inline-flex sm:ml-auto sm:mt-0 mt-2 justify-center sm:justify-start"
  }, _facebook && facebook && /*#__PURE__*/React__default.createElement("a", {
    className: "text-gray-500",
    href: "https://facebook.com/" + _facebook,
    target: "_blank"
  }, /*#__PURE__*/React__default.createElement(FacebookSvg, null)), _twitter && twitter && /*#__PURE__*/React__default.createElement("a", {
    className: "ml-3 text-gray-500",
    href: "https://twitter.com/" + _twitter,
    target: "_blank"
  }, /*#__PURE__*/React__default.createElement(TwitterSvg, null)), _instagram && instagram && /*#__PURE__*/React__default.createElement("a", {
    className: "ml-3 text-gray-500",
    href: "https://instagram.com/" + _instagram,
    target: "_blank"
  }, /*#__PURE__*/React__default.createElement(InstagramSvg, null)), _linkedin && instagram && /*#__PURE__*/React__default.createElement("a", {
    className: "ml-3 text-gray-500",
    href: "https://linkedin.com/" + _linkedin,
    target: "_blank"
  }, /*#__PURE__*/React__default.createElement(LinkedInSvg, null))))));
}

function Footer4(_ref) {
  var logo = _ref.logo,
      copyright = _ref.copyright,
      facebook = _ref.facebook,
      twitter = _ref.twitter,
      instagram = _ref.instagram,
      linkedin = _ref.linkedin,
      className = _ref.className;

  var _logo = logo || /*#__PURE__*/React__default.createElement(Logo1, null);

  var _copyright = copyright || " 2021 React-Tailblocks";

  var _facebook = facebook === undefined ? "react-tailblocks" : facebook;

  var _twitter = twitter === undefined ? "react-tailblocks" : twitter;

  var _instagram = instagram === undefined ? "react-tailblocks" : instagram;

  var _linkedin = linkedin === undefined ? "react-tailblocks" : linkedin;

  return /*#__PURE__*/React__default.createElement("footer", {
    className: "text-gray-600 body-font" + (className ? className : "")
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "container px-5 py-8 mx-auto flex items-center sm:flex-row flex-col"
  }, _logo, /*#__PURE__*/React__default.createElement("p", {
    className: "text-sm text-gray-500 sm:ml-4 sm:pl-4 sm:border-l-2 sm:border-gray-200 sm:py-2 sm:mt-0 mt-4"
  }, _copyright, _twitter && /*#__PURE__*/React__default.createElement("span", null, " -", /*#__PURE__*/React__default.createElement("a", {
    href: "https://twitter.com/" + _twitter,
    className: "text-gray-600 ml-1",
    rel: "noopener noreferrer",
    target: "_blank"
  }, "@", _twitter))), /*#__PURE__*/React__default.createElement("span", {
    className: "inline-flex sm:ml-auto sm:mt-0 mt-4 justify-center sm:justify-start"
  }, _facebook && /*#__PURE__*/React__default.createElement("a", {
    className: "text-gray-500",
    href: "https://facebook.com/" + _facebook,
    target: "_blank"
  }, /*#__PURE__*/React__default.createElement(FacebookSvg, null)), _twitter && /*#__PURE__*/React__default.createElement("a", {
    className: "ml-3 text-gray-500",
    href: "https://twitter.com/" + _twitter,
    target: "_blank"
  }, /*#__PURE__*/React__default.createElement(TwitterSvg, null)), _instagram && /*#__PURE__*/React__default.createElement("a", {
    className: "ml-3 text-gray-500",
    href: "https://instagram.com/" + _instagram,
    target: "_blank"
  }, /*#__PURE__*/React__default.createElement(InstagramSvg, null)), _linkedin && /*#__PURE__*/React__default.createElement("a", {
    className: "ml-3 text-gray-500",
    href: "https://linkedin.com/" + _linkedin,
    target: "_blank"
  }, /*#__PURE__*/React__default.createElement(LinkedInSvg, null)))));
}

function Form(_ref) {
  var children = _ref.children,
      onSubmit = _ref.onSubmit,
      className = _ref.className,
      cancelButton = _ref.cancelButton,
      saveButton = _ref.saveButton,
      editButton = _ref.editButton,
      buttons = _ref.buttons,
      options = _ref.options,
      managed = _ref.managed,
      editing = _ref.editing;

  var _managed = managed === undefined ? true : managed;

  var _options = options || {
    defaultValues: {}
  };

  var methods = reactHookForm.useForm(options);
  var handleSubmit = methods.handleSubmit;

  var _useState = React.useState(editing),
      _editing = _useState[0],
      setEditing = _useState[1];

  if (!_options.defaultValues) _options.defaultValues = {};
  React.useEffect(function () {
    setEditing(editing);
  }, [editing]);

  var _context = _extends({
    editing: _editing
  }, methods);

  return /*#__PURE__*/React__default.createElement(reactHookForm.FormProvider, _extends({}, _context, {
    className: className
  }), /*#__PURE__*/React__default.createElement("form", {
    onSubmit: handleSubmit(function (e) {
      setEditing(false);
      if (onSubmit) onSubmit(e);
    })
  }, children, _managed && /*#__PURE__*/React__default.createElement("div", {
    className: "px-4 py-3 bg-gray-50 text-right sm:px-6"
  }, /*#__PURE__*/React__default.createElement(React__default.Fragment, null, cancelButton ? cancelButton : /*#__PURE__*/React__default.createElement(React__default.Fragment, null, _editing && /*#__PURE__*/React__default.createElement("button", {
    type: "button",
    className: "mr-2 inline-flex justify-center py-2 px-4 border border-transparent shadow-sm text-sm font-medium rounded-md text-black bg-gray-300 hover:bg-gray-400 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-gray-500",
    onClick: function onClick() {
      methods.reset(_options.defaultValues);
      setEditing(false);
    }
  }, "Cancel")), saveButton ? saveButton : /*#__PURE__*/React__default.createElement(React__default.Fragment, null, _editing && /*#__PURE__*/React__default.createElement("button", {
    type: "submit",
    className: "mr-2 inline-flex justify-center py-2 px-4 border border-transparent shadow-sm text-sm font-medium rounded-md text-white bg-blue-600 hover:bg-blue-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-blue-500"
  }, "Save")), editButton ? editButton : /*#__PURE__*/React__default.createElement(React__default.Fragment, null, !_editing && /*#__PURE__*/React__default.createElement("button", {
    type: "button",
    className: "mr-2 inline-flex justify-center py-2 px-4 border border-transparent shadow-sm text-sm font-medium rounded-md text-white bg-blue-600 hover:bg-blue-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-blue-500",
    onClick: function onClick() {
      setEditing(true);
    }
  }, "Edit"))), buttons ? buttons : /*#__PURE__*/React__default.createElement(React__default.Fragment, null))));
}

function Grid1(_ref) {
  var items = _ref.items,
      className = _ref.className;

  var _items = items || [/*#__PURE__*/React__default.createElement("div", {
    className: "lg:w-1/4 md:w-1/2 p-4 w-full",
    key: "1"
  }, /*#__PURE__*/React__default.createElement("a", {
    className: "block relative h-48 rounded overflow-hidden"
  }, /*#__PURE__*/React__default.createElement("img", {
    alt: "ecommerce",
    className: "object-cover object-center w-full h-full block",
    src: "https://dummyimage.com/420x260"
  })), /*#__PURE__*/React__default.createElement("div", {
    className: "mt-4"
  }, /*#__PURE__*/React__default.createElement("h3", {
    className: "text-gray-500 text-xs tracking-widest title-font mb-1"
  }, "GAMES"), /*#__PURE__*/React__default.createElement("h2", {
    className: "text-gray-900 title-font text-lg font-medium"
  }, "Monopoly"), /*#__PURE__*/React__default.createElement("p", {
    className: "mt-1 text-sm"
  }, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce imperdiet hendrerit nisi quis viverra. Cras quis ipsum arcu. Phasellus in nibh mauris. Cras malesuada facilisis porttitor."), /*#__PURE__*/React__default.createElement("p", {
    className: "mt-2"
  }, "$24.00"))), /*#__PURE__*/React__default.createElement("div", {
    className: "lg:w-1/4 md:w-1/2 p-4 w-full",
    key: "2"
  }, /*#__PURE__*/React__default.createElement("a", {
    className: "block relative h-48 rounded overflow-hidden"
  }, /*#__PURE__*/React__default.createElement("img", {
    alt: "ecommerce",
    className: "object-cover object-center w-full h-full block",
    src: "https://dummyimage.com/420x260"
  })), /*#__PURE__*/React__default.createElement("div", {
    className: "mt-4"
  }, /*#__PURE__*/React__default.createElement("h3", {
    className: "text-gray-500 text-xs tracking-widest title-font mb-1"
  }, "GAMES"), /*#__PURE__*/React__default.createElement("h2", {
    className: "text-gray-900 title-font text-lg font-medium"
  }, "Monopoly"), /*#__PURE__*/React__default.createElement("p", {
    className: "mt-1 text-sm"
  }, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce imperdiet hendrerit nisi quis viverra. Cras quis ipsum arcu. Phasellus in nibh mauris. Cras malesuada facilisis porttitor."), /*#__PURE__*/React__default.createElement("p", {
    className: "mt-2"
  }, "$24.00"))), /*#__PURE__*/React__default.createElement("div", {
    className: "lg:w-1/4 md:w-1/2 p-4 w-full",
    key: "3"
  }, /*#__PURE__*/React__default.createElement("a", {
    className: "block relative h-48 rounded overflow-hidden"
  }, /*#__PURE__*/React__default.createElement("img", {
    alt: "ecommerce",
    className: "object-cover object-center w-full h-full block",
    src: "https://dummyimage.com/420x260"
  })), /*#__PURE__*/React__default.createElement("div", {
    className: "mt-4"
  }, /*#__PURE__*/React__default.createElement("h3", {
    className: "text-gray-500 text-xs tracking-widest title-font mb-1"
  }, "GAMES"), /*#__PURE__*/React__default.createElement("h2", {
    className: "text-gray-900 title-font text-lg font-medium"
  }, "Monopoly"), /*#__PURE__*/React__default.createElement("p", {
    className: "mt-1 text-sm"
  }, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce imperdiet hendrerit nisi quis viverra. Cras quis ipsum arcu. Phasellus in nibh mauris. Cras malesuada facilisis porttitor."), /*#__PURE__*/React__default.createElement("p", {
    className: "mt-2"
  }, "$24.00"))), /*#__PURE__*/React__default.createElement("div", {
    className: "lg:w-1/4 md:w-1/2 p-4 w-full",
    key: "4"
  }, /*#__PURE__*/React__default.createElement("a", {
    className: "block relative h-48 rounded overflow-hidden"
  }, /*#__PURE__*/React__default.createElement("img", {
    alt: "ecommerce",
    className: "object-cover object-center w-full h-full block",
    src: "https://dummyimage.com/420x260"
  })), /*#__PURE__*/React__default.createElement("div", {
    className: "mt-4"
  }, /*#__PURE__*/React__default.createElement("h3", {
    className: "text-gray-500 text-xs tracking-widest title-font mb-1"
  }, "GAMES"), /*#__PURE__*/React__default.createElement("h2", {
    className: "text-gray-900 title-font text-lg font-medium"
  }, "Monopoly"), /*#__PURE__*/React__default.createElement("p", {
    className: "mt-1 text-sm"
  }, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce imperdiet hendrerit nisi quis viverra. Cras quis ipsum arcu. Phasellus in nibh mauris. Cras malesuada facilisis porttitor."), /*#__PURE__*/React__default.createElement("p", {
    className: "mt-2"
  }, "$24.00"))), /*#__PURE__*/React__default.createElement("div", {
    className: "lg:w-1/4 md:w-1/2 p-4 w-full",
    key: "5"
  }, /*#__PURE__*/React__default.createElement("a", {
    className: "block relative h-48 rounded overflow-hidden"
  }, /*#__PURE__*/React__default.createElement("img", {
    alt: "ecommerce",
    className: "object-cover object-center w-full h-full block",
    src: "https://dummyimage.com/420x260"
  })), /*#__PURE__*/React__default.createElement("div", {
    className: "mt-4"
  }, /*#__PURE__*/React__default.createElement("h3", {
    className: "text-gray-500 text-xs tracking-widest title-font mb-1"
  }, "GAMES"), /*#__PURE__*/React__default.createElement("h2", {
    className: "text-gray-900 title-font text-lg font-medium"
  }, "Monopoly"), /*#__PURE__*/React__default.createElement("p", {
    className: "mt-1 text-sm"
  }, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce imperdiet hendrerit nisi quis viverra. Cras quis ipsum arcu. Phasellus in nibh mauris. Cras malesuada facilisis porttitor."), /*#__PURE__*/React__default.createElement("p", {
    className: "mt-2"
  }, "$24.00"))), /*#__PURE__*/React__default.createElement("div", {
    className: "lg:w-1/4 md:w-1/2 p-4 w-full",
    key: "6"
  }, /*#__PURE__*/React__default.createElement("a", {
    className: "block relative h-48 rounded overflow-hidden"
  }, /*#__PURE__*/React__default.createElement("img", {
    alt: "ecommerce",
    className: "object-cover object-center w-full h-full block",
    src: "https://dummyimage.com/420x260"
  })), /*#__PURE__*/React__default.createElement("div", {
    className: "mt-4"
  }, /*#__PURE__*/React__default.createElement("h3", {
    className: "text-gray-500 text-xs tracking-widest title-font mb-1"
  }, "GAMES"), /*#__PURE__*/React__default.createElement("h2", {
    className: "text-gray-900 title-font text-lg font-medium"
  }, "Monopoly"), /*#__PURE__*/React__default.createElement("p", {
    className: "mt-1 text-sm"
  }, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce imperdiet hendrerit nisi quis viverra. Cras quis ipsum arcu. Phasellus in nibh mauris. Cras malesuada facilisis porttitor."), /*#__PURE__*/React__default.createElement("p", {
    className: "mt-2"
  }, "$24.00"))), /*#__PURE__*/React__default.createElement("div", {
    className: "lg:w-1/4 md:w-1/2 p-4 w-full",
    key: "7"
  }, /*#__PURE__*/React__default.createElement("a", {
    className: "block relative h-48 rounded overflow-hidden"
  }, /*#__PURE__*/React__default.createElement("img", {
    alt: "ecommerce",
    className: "object-cover object-center w-full h-full block",
    src: "https://dummyimage.com/420x260"
  })), /*#__PURE__*/React__default.createElement("div", {
    className: "mt-4"
  }, /*#__PURE__*/React__default.createElement("h3", {
    className: "text-gray-500 text-xs tracking-widest title-font mb-1"
  }, "GAMES"), /*#__PURE__*/React__default.createElement("h2", {
    className: "text-gray-900 title-font text-lg font-medium"
  }, "Monopoly"), /*#__PURE__*/React__default.createElement("p", {
    className: "mt-1 text-sm"
  }, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce imperdiet hendrerit nisi quis viverra. Cras quis ipsum arcu. Phasellus in nibh mauris. Cras malesuada facilisis porttitor."), /*#__PURE__*/React__default.createElement("p", {
    className: "mt-2"
  }, "$24.00"))), /*#__PURE__*/React__default.createElement("div", {
    className: "lg:w-1/4 md:w-1/2 p-4 w-full",
    key: "8"
  }, /*#__PURE__*/React__default.createElement("a", {
    className: "block relative h-48 rounded overflow-hidden"
  }, /*#__PURE__*/React__default.createElement("img", {
    alt: "ecommerce",
    className: "object-cover object-center w-full h-full block",
    src: "https://dummyimage.com/420x260"
  })), /*#__PURE__*/React__default.createElement("div", {
    className: "mt-4"
  }, /*#__PURE__*/React__default.createElement("h3", {
    className: "text-gray-500 text-xs tracking-widest title-font mb-1"
  }, "GAMES"), /*#__PURE__*/React__default.createElement("h2", {
    className: "text-gray-900 title-font text-lg font-medium"
  }, "Monopoly"), /*#__PURE__*/React__default.createElement("p", {
    className: "mt-1 text-sm"
  }, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce imperdiet hendrerit nisi quis viverra. Cras quis ipsum arcu. Phasellus in nibh mauris. Cras malesuada facilisis porttitor."), /*#__PURE__*/React__default.createElement("p", {
    className: "mt-2"
  }, "$24.00")))];

  return /*#__PURE__*/React__default.createElement("section", {
    className: "text-gray-600 body-font " + (className ? className : "")
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "container px-5 py-24 mx-auto"
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "flex flex-wrap -m-4"
  }, _items.map(function (i) {
    return i;
  }))));
}

function Hero1(_ref) {
  var title = _ref.title,
      description = _ref.description,
      image = _ref.image,
      button1 = _ref.button1,
      button2 = _ref.button2,
      className = _ref.className;

  var _title = title || "Before they're Sold out ReadyMade Gluten";

  var _description = description || "Copper mug try-hard pitchfork pour-over freegan heirloom neutra air plant cold-pressed tacos poke beard tote bag. Heirloom echo park mlkshk tote bag selvage hot chicken authentic tumeric truffaut hexagon try-hard chambray.";

  var _image = image || "https://dummyimage.com/720x600";

  var _button1 = /*#__PURE__*/React__default.createElement("button", {
    className: "inline-flex text-white bg-blue-500 border-0 py-2 px-6 focus:outline-none hover:bg-blue-600 rounded text-lg"
  }, "Button One");

  var _button2 = /*#__PURE__*/React__default.createElement("button", {
    className: "ml-4 inline-flex text-gray-700 bg-gray-100 border-0 py-2 px-6 focus:outline-none hover:bg-gray-200 rounded text-lg"
  }, "Button Two");

  if (button1 !== undefined) {
    if (button1 && typeof button1.title === "string") {
      _button1 = /*#__PURE__*/React__default.createElement("button", {
        className: "inline-flex text-white bg-blue-500 border-0 py-2 px-6 focus:outline-none hover:bg-blue-600 rounded text-lg",
        onClick: button1.onClick ? button1.onClick : function () {}
      }, button1.title);
    } else {
      _button1 = button1;
    }
  }

  if (button2 !== undefined) {
    if (button2 && typeof button2.title === "string") {
      _button12 = /*#__PURE__*/React__default.createElement("button", {
        className: "inline-flex text-white bg-blue-500 border-0 py-2 px-6 focus:outline-none hover:bg-blue-600 rounded text-lg",
        onClick: button2.onClick ? button2.onClick : function () {}
      }, button2.title);
    } else {
      _button2 = button2;
    }
  }

  return /*#__PURE__*/React__default.createElement("section", {
    className: "text-gray-600 body-font " + (className ? className : "")
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "container mx-auto flex px-5 py-24 md:flex-row flex-col items-center"
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "lg:flex-grow md:w-1/2 lg:pr-24 md:pr-16 flex flex-col md:items-start md:text-left mb-16 md:mb-0 items-center text-center"
  }, typeof _title === "string" ? /*#__PURE__*/React__default.createElement("h1", {
    className: "title-font sm:text-4xl text-3xl mb-4 font-medium text-gray-900"
  }, _title) : _title, typeof _description === "string" ? /*#__PURE__*/React__default.createElement("p", {
    className: "mb-8 leading-relaxed"
  }, _description) : _description, /*#__PURE__*/React__default.createElement("div", {
    className: "flex justify-center"
  }, _button1, _button2)), /*#__PURE__*/React__default.createElement("div", {
    className: "lg:max-w-lg lg:w-full md:w-1/2 w-5/6"
  }, typeof _image === "string" ? /*#__PURE__*/React__default.createElement("img", {
    className: "object-cover object-center rounded",
    alt: "hero",
    src: _image
  }) : _image)));
}

function Input(_ref) {
  var _errors$name, _errors$name2;

  var name = _ref.name,
      type = _ref.type,
      prefix = _ref.prefix,
      label = _ref.label,
      value = _ref.value,
      placeHolder = _ref.placeHolder,
      className = _ref.className,
      inputClass = _ref.inputClass,
      labelClass = _ref.labelClass,
      errorClass = _ref.errorClass,
      descriptionClass = _ref.descriptionClass,
      checked = _ref.checked,
      disabled = _ref.disabled,
      options = _ref.options,
      description = _ref.description,
      _onChange = _ref.onChange,
      _onBlur = _ref.onBlur,
      _onFocus = _ref.onFocus;

  var _useFormContext = reactHookForm.useFormContext(),
      register = _useFormContext.register,
      errors = _useFormContext.formState.errors,
      getValues = _useFormContext.getValues,
      editing = _useFormContext.editing;

  var _register = register(name, options),
      onc = _register.onChange,
      onb = _register.onBlur,
      ref = _register.ref,
      nm = _register.name;

  return /*#__PURE__*/React__default.createElement("div", {
    className: "mt-2 " + (className ? className : "")
  }, type === 'checkbox' || type === "radio" ? /*#__PURE__*/React__default.createElement(React__default.Fragment, null, /*#__PURE__*/React__default.createElement("div", {
    className: "flex items-start"
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "flex items-center h-5"
  }, /*#__PURE__*/React__default.createElement("input", {
    id: name,
    name: nm,
    type: type,
    ref: ref,
    placeholder: placeHolder,
    checked: checked,
    disabled: disabled || !editing,
    onChange: function onChange(e) {
      onc(e);
      if (_onChange) _onChange(e);
    },
    onBlur: function onBlur(e) {
      onb(e);
      if (_onBlur) _onBlur(e);
    },
    onFocus: function onFocus(e) {
      if (_onFocus) _onFocus(e);
    },
    className: "focus:ring-blue-500 h-4 w-4 text-blue-600 border-gray-300 " + inputClass
  })), /*#__PURE__*/React__default.createElement("div", {
    className: "ml-3 text-sm"
  }, /*#__PURE__*/React__default.createElement("label", {
    htmlFor: name,
    className: "font-medium text-gray-700 " + labelClass
  }, label), /*#__PURE__*/React__default.createElement("p", {
    className: "text-gray-500"
  }, description))), editing && /*#__PURE__*/React__default.createElement("p", {
    className: "text-red-400 font-medium text-sm " + errorClass
  }, (_errors$name = errors[name]) === null || _errors$name === void 0 ? void 0 : _errors$name.message)) : /*#__PURE__*/React__default.createElement(React__default.Fragment, null, /*#__PURE__*/React__default.createElement("label", {
    htmlFor: name,
    className: "block text-sm font-medium text-gray-600 " + labelClass
  }, label), prefix ? /*#__PURE__*/React__default.createElement("div", {
    className: "mt-1 flex rounded-md shadow-sm"
  }, /*#__PURE__*/React__default.createElement("span", {
    className: "inline-flex items-center pl-3 pr-1 rounded-l-md border border-r-0 border-gray-300 bg-gray-50 text-gray-500 text-sm"
  }, prefix), /*#__PURE__*/React__default.createElement("input", {
    id: name,
    name: nm,
    type: type,
    ref: ref,
    value: value,
    placeholder: placeHolder,
    disabled: disabled || !editing,
    onChange: function onChange(e) {
      onc(e);
      if (_onChange) _onChange(e);
    },
    onBlur: function onBlur(e) {
      onb(e);
      if (_onBlur) _onBlur(e);
    },
    onFocus: function onFocus(e) {
      if (_onFocus) _onFocus(e);
    },
    className: "text-gray-700 p-3 focus:ring-blue-500 focus:border-blue-500 shadow-sm flex-1 block w-full rounded-none rounded-r-md sm:text-sm border border-gray-300 " + inputClass
  })) : /*#__PURE__*/React__default.createElement(React__default.Fragment, null, editing ? /*#__PURE__*/React__default.createElement("input", {
    id: name,
    name: nm,
    type: type,
    ref: ref,
    value: value,
    placeholder: placeHolder,
    disabled: disabled,
    onChange: function onChange(e) {
      onc(e);
      if (_onChange) _onChange(e);
    },
    onBlur: function onBlur(e) {
      onb(e);
      if (_onBlur) _onBlur(e);
    },
    onFocus: function onFocus(e) {
      if (_onFocus) _onFocus(e);
    },
    className: "text-gray-700 p-3 rounded-md shadow-sm focus:ring-blue-500 focus:border-blue-500 mt-1 block w-full sm:text-sm border border-gray-300 " + inputClass
  }) : /*#__PURE__*/React__default.createElement("p", null, getValues(name))), editing && /*#__PURE__*/React__default.createElement(React__default.Fragment, null, /*#__PURE__*/React__default.createElement("p", {
    className: "text-red-400 font-medium text-sm " + errorClass
  }, (_errors$name2 = errors[name]) === null || _errors$name2 === void 0 ? void 0 : _errors$name2.message), /*#__PURE__*/React__default.createElement("p", {
    className: "text-gray-400 font-medium text-sm " + descriptionClass
  }, description))));
}

function TextArea(_ref) {
  var _errors$name;

  var name = _ref.name,
      label = _ref.label,
      value = _ref.value,
      placeHolder = _ref.placeHolder,
      className = _ref.className,
      inputClass = _ref.inputClass,
      labelClass = _ref.labelClass,
      errorClass = _ref.errorClass,
      descriptionClass = _ref.descriptionClass,
      disabled = _ref.disabled,
      options = _ref.options,
      maxLength = _ref.maxLength,
      description = _ref.description,
      _onChange = _ref.onChange,
      _onBlur = _ref.onBlur,
      _onFocus = _ref.onFocus;

  var _useFormContext = reactHookForm.useFormContext(),
      register = _useFormContext.register,
      errors = _useFormContext.formState.errors,
      getValues = _useFormContext.getValues,
      editing = _useFormContext.editing;

  var _register = register(name, options),
      onc = _register.onChange,
      onb = _register.onBlur,
      ref = _register.ref,
      nm = _register.name;

  return /*#__PURE__*/React__default.createElement("div", {
    className: "mt-2 " + (className ? className : "")
  }, /*#__PURE__*/React__default.createElement("label", {
    htmlFor: name,
    className: "block text-sm font-medium text-gray-600 " + labelClass
  }, label), editing ? /*#__PURE__*/React__default.createElement(React__default.Fragment, null, /*#__PURE__*/React__default.createElement("textarea", {
    id: name,
    name: nm,
    ref: ref,
    value: value,
    placeholder: placeHolder,
    disabled: disabled,
    maxLength: maxLength,
    onChange: function onChange(e) {
      onc(e);
      if (_onChange) _onChange(e);
    },
    onBlur: function onBlur(e) {
      onb(e);
      if (_onBlur) _onBlur(e);
    },
    onFocus: function onFocus(e) {
      if (_onFocus) _onFocus(e);
    },
    className: "text-gray-700 p-3 rounded-md shadow-sm focus:ring-blue-500 focus:border-blue-500 mt-1 block w-full sm:text-sm border border-gray-300" + inputClass
  }), /*#__PURE__*/React__default.createElement("p", {
    className: "text-red-400 font-medium text-sm " + errorClass
  }, (_errors$name = errors[name]) === null || _errors$name === void 0 ? void 0 : _errors$name.message), /*#__PURE__*/React__default.createElement("p", {
    className: "text-gray-400 font-medium text-sm " + descriptionClass
  }, description)) : /*#__PURE__*/React__default.createElement("p", null, getValues(name)));
}

function Select(_ref) {
  var _errors$name;

  var name = _ref.name,
      label = _ref.label,
      className = _ref.className,
      inputClass = _ref.inputClass,
      labelClass = _ref.labelClass,
      errorClass = _ref.errorClass,
      descriptionClass = _ref.descriptionClass,
      disabled = _ref.disabled,
      options = _ref.options,
      items = _ref.items,
      size = _ref.size,
      description = _ref.description,
      _onChange = _ref.onChange,
      _onBlur = _ref.onBlur,
      _onFocus = _ref.onFocus;

  var _useFormContext = reactHookForm.useFormContext(),
      register = _useFormContext.register,
      errors = _useFormContext.formState.errors,
      getValues = _useFormContext.getValues,
      editing = _useFormContext.editing;

  var _register = register(name, options),
      onc = _register.onChange,
      onb = _register.onBlur,
      ref = _register.ref,
      nm = _register.name;

  return /*#__PURE__*/React__default.createElement("div", {
    className: "mt-2 " + (className ? className : "")
  }, /*#__PURE__*/React__default.createElement("label", {
    htmlFor: name,
    className: "block text-sm font-medium text-gray-600 " + labelClass
  }, label), editing ? /*#__PURE__*/React__default.createElement("select", {
    id: name,
    name: nm,
    ref: ref,
    disabled: disabled,
    size: size,
    onChange: function onChange(e) {
      onc(e);
      if (_onChange) _onChange(e);
    },
    onBlur: function onBlur(e) {
      onb(e);
      if (_onBlur) _onBlur(e);
    },
    onFocus: function onFocus(e) {
      if (_onFocus) _onFocus(e);
    },
    className: "text-gray-700 p-3 rounded-md shadow-sm focus:ring-blue-500 focus:border-blue-500 mt-1 block w-full sm:text-sm border border-gray-300" + inputClass
  }, items.map(function (o, i) {
    o.selected = getValues(name) === o.title;
    return /*#__PURE__*/React__default.createElement("option", _extends({
      key: i
    }, o), o.title);
  })) : /*#__PURE__*/React__default.createElement("p", null, getValues(name)), /*#__PURE__*/React__default.createElement("p", {
    className: "text-red-400 font-medium text-sm " + errorClass
  }, (_errors$name = errors[name]) === null || _errors$name === void 0 ? void 0 : _errors$name.message), /*#__PURE__*/React__default.createElement("p", {
    className: "text-gray-400 font-medium text-sm " + descriptionClass
  }, description));
}

function Form1(_ref) {
  var className = _ref.className,
      panels = _ref.panels,
      auto = _ref.auto,
      collapsible = _ref.collapsible,
      summary = _ref.summary,
      divider = _ref.divider,
      onCollapse = _ref.onCollapse;

  var _collapsible = collapsible === undefined ? true : collapsible;

  var _auto = auto === undefined ? true : auto;

  var _summary = summary === undefined ? true : summary;

  var _divider = divider || /*#__PURE__*/React__default.createElement("div", {
    className: "hidden sm:block",
    "aria-hidden": "true"
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "py-6"
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "border-t border-gray-200"
  })));

  var _panels = panels || [{
    label: /*#__PURE__*/React__default.createElement("h3", {
      className: "text-md font-medium leading-6 text-gray-600"
    }, "Profile"),
    title: /*#__PURE__*/React__default.createElement("h3", {
      className: "text-lg font-medium leading-6 text-gray-900"
    }, "Profile"),
    description: /*#__PURE__*/React__default.createElement("p", {
      className: "mt-1 text-sm text-gray-600"
    }, "This information will be displayed publicly so be careful what you share."),
    content: /*#__PURE__*/React__default.createElement(Form, {
      onSubmit: function onSubmit(data) {
        console.log(data);
      }
    }, /*#__PURE__*/React__default.createElement("div", {
      className: "px-4 sm:p-6"
    }, /*#__PURE__*/React__default.createElement("div", {
      className: "grid grid-cols-3 gap-6"
    }, /*#__PURE__*/React__default.createElement("div", {
      className: "col-span-3 sm:col-span-2"
    }, /*#__PURE__*/React__default.createElement(Input, {
      name: "website",
      label: "Website",
      prefix: "http://",
      placeHolder: "www.example.com"
    }))), /*#__PURE__*/React__default.createElement(TextArea, {
      name: "about",
      label: "About",
      placeHolder: "All about you",
      description: "Brief description for your profile. URLs are hyperlinked."
    }), /*#__PURE__*/React__default.createElement("div", {
      className: "mt-4"
    }, /*#__PURE__*/React__default.createElement("label", {
      className: "block text-sm font-medium text-gray-700"
    }, "Photo"), /*#__PURE__*/React__default.createElement("div", {
      className: "mt-1 flex items-center"
    }, /*#__PURE__*/React__default.createElement("span", {
      className: "inline-block h-12 w-12 rounded-full overflow-hidden bg-gray-100"
    }, /*#__PURE__*/React__default.createElement(outline.UserCircleIcon, {
      className: "h-full w-full text-gray-300"
    })), /*#__PURE__*/React__default.createElement("button", {
      type: "button",
      className: "ml-5 bg-white py-2 px-3 border border-gray-300 rounded-md shadow-sm text-sm leading-4 font-medium text-gray-700 hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-blue-500"
    }, "Change"))), /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement("label", {
      className: "block text-sm font-medium text-gray-700"
    }, "Cover photo"), /*#__PURE__*/React__default.createElement("div", {
      className: "mt-1 flex justify-center px-6 pt-5 pb-6 border-2 border-gray-300 border-dashed rounded-md"
    }, /*#__PURE__*/React__default.createElement("div", {
      className: "space-y-1 text-center"
    }, /*#__PURE__*/React__default.createElement(outline.PhotographIcon, {
      className: "mx-auto h-12 w-12 text-gray-400"
    }), /*#__PURE__*/React__default.createElement("div", {
      className: "flex text-sm text-gray-600"
    }, /*#__PURE__*/React__default.createElement("label", {
      htmlFor: "file-upload",
      className: "relative cursor-pointer bg-white rounded-md font-medium text-blue-600 hover:text-blue-500 focus-within:outline-none focus-within:ring-2 focus-within:ring-offset-2 focus-within:ring-blue-500"
    }, /*#__PURE__*/React__default.createElement("span", null, "Upload a file"), /*#__PURE__*/React__default.createElement("input", {
      id: "file-upload",
      name: "file-upload",
      type: "file",
      className: "sr-only"
    })), /*#__PURE__*/React__default.createElement("p", {
      className: "pl-1"
    }, "or drag and drop")), /*#__PURE__*/React__default.createElement("p", {
      className: "text-xs text-gray-500"
    }, "PNG, JPG, GIF up to 10MB"))))))
  }, {
    label: /*#__PURE__*/React__default.createElement("h3", {
      className: "text-md font-medium leading-6 text-gray-600"
    }, "Personal Information"),
    title: /*#__PURE__*/React__default.createElement("h3", {
      className: "text-lg font-medium leading-6 text-gray-900"
    }, "Personal Information"),
    description: /*#__PURE__*/React__default.createElement("p", {
      className: "mt-1 text-sm text-gray-600"
    }, "Use a permanent address where you can receive mail."),
    content: /*#__PURE__*/React__default.createElement(Form, {
      onSubmit: function onSubmit(data) {
        console.log(data);
      },
      options: {
        defaultValues: {
          firstName: "Mike",
          lastName: "Smith",
          email: "mike@domain.com",
          country: "United States",
          street: "123 N Place",
          city: "Rock Spring",
          state: "WY",
          postal: "82901"
        }
      }
    }, /*#__PURE__*/React__default.createElement("div", {
      className: "grid grid-cols-6 gap-6 px-4 sm:p-6"
    }, /*#__PURE__*/React__default.createElement("div", {
      className: "col-span-6 sm:col-span-3"
    }, /*#__PURE__*/React__default.createElement(Input, {
      name: "firstName",
      label: "First name"
    })), /*#__PURE__*/React__default.createElement("div", {
      className: "col-span-6 sm:col-span-3"
    }, /*#__PURE__*/React__default.createElement(Input, {
      name: "lastName",
      label: "Last name"
    })), /*#__PURE__*/React__default.createElement("div", {
      className: "col-span-6 sm:col-span-4"
    }, /*#__PURE__*/React__default.createElement(Input, {
      name: "email",
      type: "email",
      label: "Email address",
      options: {
        required: "This field is required"
      },
      description: "Please be sure and enter your personal email address.",
      placeHolder: "user@domain.com"
    })), /*#__PURE__*/React__default.createElement("div", {
      className: "col-span-6 sm:col-span-3"
    }, /*#__PURE__*/React__default.createElement(Select, {
      name: "country",
      label: "Contry/Region",
      items: [{
        title: "United States"
      }, {
        title: "Canada"
      }, {
        title: "Mexico"
      }]
    })), /*#__PURE__*/React__default.createElement("div", {
      className: "col-span-6"
    }, /*#__PURE__*/React__default.createElement(Input, {
      name: "street",
      label: "Street address"
    })), /*#__PURE__*/React__default.createElement("div", {
      className: "col-span-6 sm:col-span-6 lg:col-span-2"
    }, /*#__PURE__*/React__default.createElement(Input, {
      name: "city",
      label: "City"
    })), /*#__PURE__*/React__default.createElement("div", {
      className: "col-span-6 sm:col-span-3 lg:col-span-2"
    }, /*#__PURE__*/React__default.createElement(Input, {
      name: "state",
      label: "State/Province"
    })), /*#__PURE__*/React__default.createElement("div", {
      className: "col-span-6 sm:col-span-3 lg:col-span-2"
    }, /*#__PURE__*/React__default.createElement(Input, {
      name: "postal",
      label: "Zip / Postal"
    }))))
  }, {
    label: /*#__PURE__*/React__default.createElement("h3", {
      className: "text-md font-medium leading-6 text-gray-600"
    }, "Notifications"),
    title: /*#__PURE__*/React__default.createElement("h3", {
      className: "text-lg font-medium leading-6 text-gray-900"
    }, "Notifications"),
    description: /*#__PURE__*/React__default.createElement("p", {
      className: "mt-1 text-sm text-gray-600"
    }, "Decide which communications you'd like to receive and how."),
    content: /*#__PURE__*/React__default.createElement(Form, {
      onSubmit: function onSubmit(data) {
        console.log(data);
      },
      options: {
        defaultValues: {
          comments: true,
          candidates: false,
          offers: true
        }
      }
    }, /*#__PURE__*/React__default.createElement("div", {
      className: "px-4 sm:p-6"
    }, /*#__PURE__*/React__default.createElement("fieldset", null, /*#__PURE__*/React__default.createElement("legend", {
      className: "text-base font-medium text-gray-900"
    }, "By Email"), /*#__PURE__*/React__default.createElement("div", {
      className: "mt-4 space-y-4"
    }, /*#__PURE__*/React__default.createElement(Input, {
      name: "comments",
      type: "checkbox",
      label: "Comments",
      description: "Get notified when someone posts a comment."
    }), /*#__PURE__*/React__default.createElement(Input, {
      name: "candidates",
      type: "checkbox",
      label: "Candidates",
      description: "Get notified when a candidate applies for a job."
    }), /*#__PURE__*/React__default.createElement(Input, {
      name: "offers",
      type: "checkbox",
      label: "Offers",
      description: "Get notified when a candidate accepts or rejects an offer."
    }))), /*#__PURE__*/React__default.createElement("fieldset", null, /*#__PURE__*/React__default.createElement("div", {
      className: "pt-8"
    }, /*#__PURE__*/React__default.createElement("legend", {
      className: "text-base font-medium text-gray-900"
    }, "Push Notifications"), /*#__PURE__*/React__default.createElement("p", {
      className: "text-sm text-gray-500"
    }, "These are delivered via SMS to your mobile phone.")), /*#__PURE__*/React__default.createElement("div", {
      className: "mt-4 space-y-4"
    }, /*#__PURE__*/React__default.createElement(Input, {
      name: "pushEverything",
      type: "radio",
      label: "Everything"
    }), /*#__PURE__*/React__default.createElement(Input, {
      name: "pushEmail",
      type: "radio",
      label: "Same as email"
    }), /*#__PURE__*/React__default.createElement(Input, {
      name: "pushNothing",
      type: "radio",
      label: "No push notifications"
    })))))
  }];

  var items = _panels.map(function () {
    return !_collapsible ? false : _auto;
  });

  if (_auto) items[0] = false;

  var _useState = React.useState(items),
      collapsed = _useState[0],
      setCollapsed = _useState[1];

  return /*#__PURE__*/React__default.createElement("div", {
    className: "bg-gray-100 " + className
  }, _panels.map(function (p, i) {
    return /*#__PURE__*/React__default.createElement(React__default.Fragment, {
      key: i
    }, /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement("div", {
      className: "md:grid md:gap-6 " + (_summary ? "md:grid-cols-3" : "md:grid-cols-2")
    }, _summary && /*#__PURE__*/React__default.createElement("div", {
      className: "md:col-span-1"
    }, /*#__PURE__*/React__default.createElement("div", {
      className: "px-4 sm:px-0"
    }, typeof p.title === 'string' ? /*#__PURE__*/React__default.createElement("h3", {
      className: "text-lg font-medium leading-6 text-gray-900"
    }, p.title) : p.title, typeof p.description === 'string' ? /*#__PURE__*/React__default.createElement("p", {
      className: "mt-1 text-sm text-gray-600"
    }, p.description) : p.description)), /*#__PURE__*/React__default.createElement("div", {
      className: "mt-5 md:mt-0 md:col-span-2"
    }, /*#__PURE__*/React__default.createElement("div", {
      className: "shadow overflow-hidden sm:rounded-md"
    }, /*#__PURE__*/React__default.createElement("div", {
      className: "bg-white"
    }, _collapsible && /*#__PURE__*/React__default.createElement("div", {
      className: "p-4 w-full flex border-b border-gray-100 " + (p.label ? "justify-between" : "justify-end")
    }, p.label, collapsed[i] ? /*#__PURE__*/React__default.createElement(outline.ChevronDownIcon, {
      className: "h-4 w-4 text-gray-400 cursor-pointer",
      onClick: function onClick() {
        var c = !_auto ? collapsed.map(function (x) {
          return x;
        }) : _panels.map(function () {
          return true;
        });
        c[i] = false;
        setCollapsed(c);
        if (onCollapse) onCollapse(c);
      }
    }) : /*#__PURE__*/React__default.createElement(outline.ChevronUpIcon, {
      className: "h-4 w-4 text-gray-400 cursor-pointer",
      onClick: function onClick() {
        var c = !_auto ? collapsed.map(function (x) {
          return x;
        }) : _panels.map(function () {
          return true;
        });
        c[i] = true;
        setCollapsed(c);
        if (onCollapse) onCollapse(c);
      }
    })), !collapsed[i] && p.content))))), _divider);
  }));
}

function Login1(_ref) {
  var title = _ref.title,
      description = _ref.description,
      logo = _ref.logo,
      remember = _ref.remember,
      forgotUrl = _ref.forgotUrl,
      registerUrl = _ref.registerUrl,
      onLogin = _ref.onLogin,
      action = _ref.action,
      passwordLength = _ref.passwordLength,
      error = _ref.error,
      hidden = _ref.hidden;

  var _useState = React.useState(null),
      email = _useState[0],
      setEmail = _useState[1];

  var _useState2 = React.useState(null),
      password = _useState2[0],
      setPassword = _useState2[1];

  var _useState3 = React.useState(false),
      rememberMe = _useState3[0],
      setRememberMe = _useState3[1];

  var _useState4 = React.useState(error),
      errorMsg = _useState4[0],
      setErrorMsg = _useState4[1];

  var _pwlen = passwordLength || 1;

  var _remember = remember == undefined ? true : remember;

  var _forgot = forgotUrl;
  var _register = registerUrl;

  var _title = title || "Sign Into Your Account";

  var _description = description || "By signing into your account you get access to all the awesome features our system has to offer.";

  var _logo = logo || /*#__PURE__*/React__default.createElement(outline.CubeIcon, {
    className: "p-2 mx-auto w-16 h-16 text-white bg-blue-500 rounded-full"
  });

  React.useEffect(function () {
    setErrorMsg(error);
  }, [error]);

  var isValid = function isValid() {
    setErrorMsg(null);

    if (email && password) {
      if (password.length >= _pwlen && email.includes("@") && email.includes(".")) return true;
      setErrorMsg("Password must have at least " + _pwlen + " characters and the email address must be in a valid format.");
      return false;
    }

    setErrorMsg("Email and password are required.");
    return false;
  };

  return /*#__PURE__*/React__default.createElement("div", {
    className: "min-h-screen flex items-center justify-center bg-gray-50 py-12 px-4 sm:px-6 lg:px-8"
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "max-w-md w-full space-y-8"
  }, /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement("div", {
    className: "mx-auto"
  }, _logo), /*#__PURE__*/React__default.createElement("h2", {
    className: "mt-6 text-center text-3xl font-medium text-gray-900"
  }, _title), /*#__PURE__*/React__default.createElement("h4", {
    className: "mt-6 text-center text-sm text-gray-500"
  }, _description)), /*#__PURE__*/React__default.createElement("form", {
    className: "mt-8 space-y-6",
    method: "POST",
    action: action
  }, hidden && hidden.map(function (h) {
    return h;
  }), /*#__PURE__*/React__default.createElement("div", {
    className: "rounded-md shadow-sm -space-y-px"
  }, /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement("label", {
    htmlFor: "email-address",
    className: "sr-only"
  }, "Email address"), /*#__PURE__*/React__default.createElement("input", {
    id: "username",
    name: "username",
    type: "email",
    autoComplete: "email",
    required: true,
    className: "appearance-none rounded-none relative block w-full px-3 py-2 border  border-gray-300 placeholder-gray-500 text-gray-900 rounded-t-md focus:outline-none focus:ring-blue-500 focus:border-blue-500 focus:z-10 sm:text-sm",
    placeholder: "Email address",
    onChange: function onChange(e) {
      setEmail(e.target.value);
    }
  })), /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement("label", {
    htmlFor: "password",
    className: "sr-only"
  }, "Password"), /*#__PURE__*/React__default.createElement("input", {
    id: "password",
    name: "password",
    type: "password",
    autoComplete: "current-password",
    required: true,
    className: "appearance-none rounded-none relative block w-full px-3  py-2 border border-gray-300 placeholder-gray-500 text-gray-900 rounded-b-md focus:outline-none focus:ring-blue-500 focus:border-blue-500 focus:z-10  sm:text-sm",
    placeholder: "Password",
    onChange: function onChange(e) {
      setPassword(e.target.value);
    }
  }))), _remember && /*#__PURE__*/React__default.createElement("div", {
    className: "flex items-center"
  }, /*#__PURE__*/React__default.createElement("input", {
    id: "remember_me",
    name: "remember_me",
    type: "checkbox",
    className: "h-4 w-4 text-blue-600 focus:ring-blue-500 border-gray-300 rounded",
    checked: rememberMe,
    onChange: function onChange() {
      setRememberMe(!rememberMe);
    }
  }), /*#__PURE__*/React__default.createElement("label", {
    htmlFor: "remember_me",
    className: "ml-2 block text-sm text-gray-900"
  }, "Remember me")), /*#__PURE__*/React__default.createElement("div", {
    className: "flex items-center justify-between"
  }, _forgot && /*#__PURE__*/React__default.createElement("div", {
    className: "text-sm"
  }, /*#__PURE__*/React__default.createElement("a", {
    href: _forgot,
    className: "font-medium text-blue-600 hover:text-blue-500"
  }, "Forgot your password?")), _register && /*#__PURE__*/React__default.createElement("div", {
    className: "text-sm"
  }, /*#__PURE__*/React__default.createElement("a", {
    href: _register,
    className: "font-medium text-blue-600 hover:text-blue-500"
  }, "Register for an account?"))), errorMsg && /*#__PURE__*/React__default.createElement("div", {
    className: "text-red-500"
  }, errorMsg), /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement("button", {
    type: action ? "submit" : "button",
    className: "group relative w-full flex justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-md text-white bg-blue-600  hover:bg-blue-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-blue-500",
    onClick: function onClick() {
      if (onLogin && isValid()) onLogin({
        username: email,
        password: password,
        remember: rememberMe
      });
    }
  }, /*#__PURE__*/React__default.createElement("span", {
    className: "absolute left-0 inset-y-0 flex items-center pl-3"
  }, /*#__PURE__*/React__default.createElement(outline.LockClosedIcon, {
    className: "h-5 w-5 text-blue-500 group-hover:text-blue-400"
  })), "Sign in")))));
}

function Modal1(_ref) {
  var className = _ref.className,
      show = _ref.show,
      header = _ref.header,
      icon = _ref.icon,
      content = _ref.content,
      footer = _ref.footer,
      useForm = _ref.useForm;

  var _header = header || /*#__PURE__*/React__default.createElement("h3", {
    className: "text-lg leading-6 font-medium text-gray-900",
    id: "modal-title"
  }, "Deactivate Account");

  var _icon = icon || /*#__PURE__*/React__default.createElement("div", {
    className: "mx-auto flex-shrink-0 flex items-center justify-center h-12 w-12 rounded-full bg-red-100 sm:mx-0 sm:h-10 sm:w-10"
  }, /*#__PURE__*/React__default.createElement(outline.ExclamationIcon, {
    className: "h-6 w-6 text-red-600"
  }));

  var _content = content || /*#__PURE__*/React__default.createElement("p", {
    className: "text-sm text-gray-500"
  }, "Are you sure you want to deactivate your account? All of your data will be permanently removed. This action cannot be undone.");

  var _footer = footer || /*#__PURE__*/React__default.createElement(React__default.Fragment, null, /*#__PURE__*/React__default.createElement("button", {
    type: "button",
    className: "w-full inline-flex justify-center rounded-md border border-transparent shadow-sm px-4 py-2 bg-red-600 text-base font-medium text-white hover:bg-red-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-red-500 sm:ml-3 sm:w-auto sm:text-sm"
  }, "Deactivate"), /*#__PURE__*/React__default.createElement("button", {
    type: "button",
    className: "mt-3 w-full inline-flex justify-center rounded-md border border-gray-300 shadow-sm px-4 py-2 bg-white text-base font-medium text-gray-700 hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500 sm:mt-0 sm:ml-3 sm:w-auto sm:text-sm"
  }, "Cancel"));

  var children = /*#__PURE__*/React__default.createElement(React__default.Fragment, null, /*#__PURE__*/React__default.createElement("div", {
    className: "fixed inset-0 bg-gray-500 bg-opacity-75 transition-opacity",
    "aria-hidden": "true"
  }), /*#__PURE__*/React__default.createElement("span", {
    className: "hidden sm:inline-block sm:align-middle sm:h-screen",
    "aria-hidden": "true"
  }, "\u200B"), /*#__PURE__*/React__default.createElement("div", {
    className: "inline-block align-bottom bg-white rounded-lg text-left overflow-hidden shadow-xl transform transition-all sm:my-8 sm:align-middle sm:max-w-lg sm:w-full"
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "bg-white px-4 pt-5 pb-4 sm:p-6 sm:pb-4"
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "sm:flex sm:items-start"
  }, _icon, /*#__PURE__*/React__default.createElement("div", {
    className: "mt-3 text-center w-full sm:mt-0 sm:ml-4 sm:text-left"
  }, _header, /*#__PURE__*/React__default.createElement("div", {
    className: "mt-2"
  }, _content)))), /*#__PURE__*/React__default.createElement("div", {
    className: "bg-gray-50 px-4 py-3 sm:px-6 sm:flex sm:flex-row-reverse"
  }, _footer)));
  return /*#__PURE__*/React__default.createElement(React__default.Fragment, null, show && /*#__PURE__*/React__default.createElement("div", {
    className: "fixed z-10 inset-0 overflow-y-auto " + (className ? className : ""),
    "aria-labelledby": "modal-title",
    role: "dialog",
    "aria-modal": "true"
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "flex items-end justify-center min-h-screen pt-4 px-4 pb-20 text-center sm:block sm:p-0"
  }, useForm ? /*#__PURE__*/React__default.createElement("form", {
    onSubmit: function onSubmit(e) {
      if (_onSubmit) return _onSubmit(e);
    }
  }, children) : children)));
}

function Pricing1(_ref) {
  var title = _ref.title,
      description = _ref.description,
      categories = _ref.categories,
      starting = _ref.starting,
      interval = _ref.interval,
      cards = _ref.cards,
      className = _ref.className,
      color = _ref.color;

  var _bdcolor = color ? {
    borderColor: color
  } : {};

  var _bgcolor = color ? {
    backgroundColor: color
  } : {};

  var _interval = interval === undefined ? true : interval;

  var _title = title || "Pricing";

  var _description = description || "Whatever cardigan tote bag tumblr hexagon brooklyn asymmetrical.";

  var _categories = categories || ["Monthly", "Annually"];

  var _starting = starting || _categories[0];

  var _cards = cards || [{
    title: "START",
    description: "Literally you probably haven't heard of them jean shorts",
    price: "Free",
    features: ["Vexillologist pitchfork", "Tumeric plaid portland", "Mixtape chillwave tumeric"],
    button: {
      title: "Button",
      onClick: function onClick() {
        alert("Button 'Free' clicked");
      }
    },
    image: "https://dummyimage.com/720x400",
    category: "Monthly",
    disabled: true
  }, {
    title: "START",
    description: "Literally you probably haven't heard of them jean shorts",
    price: "Free",
    features: ["Vexillologist pitchfork", "Tumeric plaid portland", "Mixtape chillwave tumeric"],
    button: {
      title: "Button",
      onClick: function onClick() {
        alert("Button 'Free' clicked");
      }
    },
    image: "https://dummyimage.com/720x400",
    category: "Annually",
    disabled: true
  }, {
    title: "PRO",
    description: "Literally you probably haven't heard of them jean shorts",
    price: "$38",
    unit: "mo",
    popular: "POPULAR",
    features: ["Vexillologist pitchfork", "Tumeric plaid portland", "Mixtape chillwave tumeric", "Hexagon neurtra unicorn"],
    button: {
      title: "Button",
      onClick: function onClick() {
        alert("Button '$38' clicked");
      }
    },
    image: "https://dummyimage.com/720x400",
    category: "Monthly"
  }, {
    title: "PRO",
    description: "Literally you probably haven't heard of them jean shorts",
    price: "$380",
    unit: "yr",
    popular: "POPULAR",
    features: ["Vexillologist pitchfork", "Tumeric plaid portland", "Mixtape chillwave tumeric", "Hexagon neurtra unicorn"],
    button: {
      title: "Button",
      onClick: function onClick() {
        alert("Button '$380' clicked");
      }
    },
    image: "https://dummyimage.com/720x400",
    category: "Annually"
  }, {
    title: "BUSINESS",
    description: "Literally you probably haven't heard of them jean shorts",
    price: "$56",
    unit: "mo",
    features: ["Vexillologist pitchfork", "Tumeric plaid portland", "Mixtape chillwave tumeric", "Hexagon neurtra unicorn", "Happenstance arrugla bowl"],
    button: {
      title: "Button",
      onClick: function onClick() {
        alert("Button '$56' clicked");
      }
    },
    image: "https://dummyimage.com/720x400",
    category: "Monthly"
  }, {
    title: "BUSINESS",
    description: "Literally you probably haven't heard of them jean shorts",
    price: "$560",
    unit: "yr",
    features: ["Vexillologist pitchfork", "Tumeric plaid portland", "Mixtape chillwave tumeric", "Hexagon neurtra unicorn", "Happenstance arrugla bowl"],
    button: {
      title: "Button",
      onClick: function onClick() {
        alert("Button '$560' clicked");
      }
    },
    image: "https://dummyimage.com/720x400",
    category: "Annually"
  }, {
    title: "SPECIAL",
    description: "Literally you probably haven't heard of them jean shorts",
    price: "$72",
    unit: "mo",
    features: ["Vexillologist pitchfork", "Tumeric plaid portland", "Mixtape chillwave tumeric", "Hexagon neurtra unicorn", "Happenstance arrugla bowl"],
    button: {
      title: "Button",
      onClick: function onClick() {
        alert("Button '$72' clicked");
      }
    },
    image: "https://dummyimage.com/720x400",
    category: "Monthly"
  }, {
    title: "SPECIAL",
    description: "Literally you probably haven't heard of them jean shorts",
    price: "$720",
    unit: "yr",
    features: ["Vexillologist pitchfork", "Tumeric plaid portland", "Mixtape chillwave tumeric", "Hexagon neurtra unicorn", "Happenstance arrugla bowl"],
    button: {
      title: "Button",
      onClick: function onClick() {
        alert("Button '$720' clicked");
      }
    },
    image: "https://dummyimage.com/720x400",
    category: "Annually"
  }];

  var _useState = React.useState(_starting),
      category = _useState[0],
      setCategory = _useState[1];

  return /*#__PURE__*/React__default.createElement("section", {
    className: "text-gray-600 body-font overflow-hidden " + (className ? className : "")
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "container px-5 py-24 w-full mx-auto"
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "flex flex-col text-center w-full mb-20"
  }, typeof _title === 'string' ? /*#__PURE__*/React__default.createElement("h1", {
    className: "sm:text-4xl text-3xl font-medium title-font mb-2 text-gray-900"
  }, _title) : _title, typeof _description === 'string' ? /*#__PURE__*/React__default.createElement("p", {
    className: "lg:w-2/3 mx-auto leading-relaxed text-base text-gray-500"
  }, _description) : _description, _interval && /*#__PURE__*/React__default.createElement("div", {
    className: "flex mx-auto border-2 border-blue-500 rounded overflow-hidden mt-6",
    style: _bdcolor
  }, _categories.map(function (c, i) {
    var focus = c === category ? "bg-blue-500 text-white" : "";
    var sfocus = c === category ? _bgcolor : {};
    return /*#__PURE__*/React__default.createElement("button", {
      className: "py-1 px-4 focus:outline-none " + focus,
      style: sfocus,
      key: i,
      onClick: function onClick() {
        return setCategory(c);
      }
    }, c);
  }))), /*#__PURE__*/React__default.createElement("div", {
    className: "flex justify-center flex-wrap -m-4"
  }, _cards.filter(function (c) {
    if (c.category === category) return c;
  }).map(function (c, i) {
    var popular = c.popular ? "border-blue-500" : "";
    var spopular = c.popular ? _bdcolor : {};
    return /*#__PURE__*/React__default.createElement("div", {
      className: "p-4 xl:w-1/4 md:w-1/2 w-full",
      key: i
    }, /*#__PURE__*/React__default.createElement("div", {
      className: "h-full p-6 rounded-lg border-2 border-gray-300 flex flex-col relative overflow-hidden " + popular,
      style: spopular
    }, popular !== "" && /*#__PURE__*/React__default.createElement("span", {
      className: "bg-blue-500 text-white px-3 py-1 tracking-widest text-xs absolute right-0 top-0 rounded-bl",
      style: _bgcolor
    }, c.popular), typeof c.title === 'string' ? /*#__PURE__*/React__default.createElement("h2", {
      className: "text-sm tracking-widest title-font mb-1 font-medium"
    }, c.title) : c.title, typeof c.price === 'string' ? /*#__PURE__*/React__default.createElement("h1", {
      className: "text-5xl text-gray-900 leading-none flex items-center pb-4 mb-4 border-b border-gray-200"
    }, /*#__PURE__*/React__default.createElement("span", null, c.price), c.unit && /*#__PURE__*/React__default.createElement("span", {
      className: "text-lg ml-1 font-normal text-gray-500"
    }, "/", c.unit)) : c.price, c.disabled && /*#__PURE__*/React__default.createElement("p", {
      className: "text-sm text-gray-800 mb-3"
    }, "This tier is not available."), typeof c.image === "string" ? /*#__PURE__*/React__default.createElement("img", {
      className: "lg:h-48 md:h-36 w-full mb-4 object-cover object-center",
      src: c.image,
      alt: "tier"
    }) : c.image !== undefined && c.image, /*#__PURE__*/React__default.createElement("div", {
      className: "mb-4"
    }, c.features.map(function (f, i) {
      var tag = typeof f === 'string' ? /*#__PURE__*/React__default.createElement("p", {
        className: "flex items-center text-gray-600 mb-2",
        key: i
      }, /*#__PURE__*/React__default.createElement("span", {
        className: "w-4 h-4 mr-2 inline-flex items-center justify-center bg-gray-400 text-white rounded-full flex-shrink-0"
      }, /*#__PURE__*/React__default.createElement(outline.CheckIcon, {
        className: "w-3 h-3"
      })), " ", f) : f;
      return tag;
    })), typeof c.button.title === 'string' ? /*#__PURE__*/React__default.createElement("button", {
      disabled: c.disabled,
      className: "flex items-center mt-auto text-white border-0 py-2 px-4 w-full focus:outline-none " + (c.disabled ? "bg-gray-300 " : "bg-gray-400 hover:bg-gray-500") + " rounded " + (popular === "" ? '' : "bg-blue-500"),
      style: popular === "" ? {} : _bgcolor,
      onClick: c.button.onClick ? c.button.onClick : function () {}
    }, c.button.title, /*#__PURE__*/React__default.createElement(outline.ArrowSmRightIcon, {
      className: "ml-4 h-5 w-5"
    })) : c.button, /*#__PURE__*/React__default.createElement("p", {
      className: "text-xs text-gray-500 mt-3"
    }, c.description)));
  }))));
}

function Register1(_ref) {
  var title = _ref.title,
      description = _ref.description,
      logo = _ref.logo,
      loginUrl = _ref.loginUrl,
      onRegister = _ref.onRegister,
      passwordLength = _ref.passwordLength,
      action = _ref.action,
      error = _ref.error;

  var _useState = React.useState(null),
      email = _useState[0],
      setEmail = _useState[1];

  var _useState2 = React.useState(null),
      password = _useState2[0],
      setPassword = _useState2[1];

  var _useState3 = React.useState(error),
      errorMsg = _useState3[0],
      setErrorMsg = _useState3[1];

  var _pwlen = passwordLength || 1;

  var _loginUrl = loginUrl;

  var _title = title || "Register for an Account";

  var _description = description || "By registering for a free account you get access to all the awesome features our system has to offer.";

  var _logo = logo || /*#__PURE__*/React__default.createElement(outline.CubeIcon, {
    className: "p-2 mx-auto w-16 h-16 text-white bg-blue-500 rounded-full"
  });

  React.useEffect(function () {
    setErrorMsg(error);
  }, [error]);

  var isValid = function isValid() {
    setErrorMsg(null);

    if (email && password) {
      if (password.length >= _pwlen && email.includes("@") && email.includes(".")) return true;
      setErrorMsg("Password must have at least " + _pwlen + " characters and the email address must be in a valid format.");
      return false;
    }

    setErrorMsg("Email and password are required.");
    return false;
  };

  return /*#__PURE__*/React__default.createElement("div", {
    className: "min-h-screen flex items-center justify-center bg-gray-50 py-12 px-4 sm:px-6 lg:px-8"
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "max-w-md w-full space-y-8"
  }, /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement("div", {
    className: "mx-auto"
  }, _logo), /*#__PURE__*/React__default.createElement("h2", {
    className: "mt-6 text-center text-3xl font-medium text-gray-900"
  }, _title), /*#__PURE__*/React__default.createElement("h4", {
    className: "mt-6 text-center text-sm text-gray-500"
  }, _description)), /*#__PURE__*/React__default.createElement("form", {
    className: "mt-8 space-y-6",
    method: "POST",
    action: action
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "rounded-md shadow-sm -space-y-px"
  }, /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement("label", {
    htmlFor: "email-address",
    className: "sr-only"
  }, "Email address"), /*#__PURE__*/React__default.createElement("input", {
    id: "username",
    name: "username",
    type: "email",
    autoComplete: "email",
    required: true,
    className: "appearance-none rounded-none relative block w-full px-3 py-2 border  border-gray-300 placeholder-gray-500 text-gray-900 rounded-t-md focus:outline-none focus:ring-blue-500 focus:border-blue-500 focus:z-10 sm:text-sm",
    placeholder: "Email address",
    onChange: function onChange(e) {
      setEmail(e.target.value);
    }
  })), /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement("label", {
    htmlFor: "password",
    className: "sr-only"
  }, "Password"), /*#__PURE__*/React__default.createElement("input", {
    id: "password",
    name: "password",
    type: "password",
    autoComplete: "current-password",
    required: true,
    className: "appearance-none rounded-none relative block w-full px-3  py-2 border border-gray-300 placeholder-gray-500 text-gray-900 rounded-b-md focus:outline-none focus:ring-blue-500 focus:border-blue-500 focus:z-10  sm:text-sm",
    placeholder: "Password",
    onChange: function onChange(e) {
      setPassword(e.target.value);
    }
  }))), /*#__PURE__*/React__default.createElement("div", {
    className: "flex items-center justify-center"
  }, _loginUrl && /*#__PURE__*/React__default.createElement("div", {
    className: "text-sm"
  }, /*#__PURE__*/React__default.createElement("a", {
    href: _loginUrl,
    className: "font-medium text-blue-600 hover:text-blue-500"
  }, "Sign in instead?"))), errorMsg && /*#__PURE__*/React__default.createElement("div", {
    className: "text-red-500"
  }, errorMsg), /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement("button", {
    type: action ? "submit" : "button",
    className: "group relative w-full flex justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-md text-white bg-blue-600  hover:bg-blue-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-blue-500",
    onClick: function onClick() {
      if (onRegister && isValid()) onRegister({
        username: email,
        password: password
      });
    }
  }, /*#__PURE__*/React__default.createElement("span", {
    className: "absolute left-0 inset-y-0 flex items-center pl-3"
  }, /*#__PURE__*/React__default.createElement(outline.LockClosedIcon, {
    className: "h-5 w-5 text-blue-500 group-hover:text-blue-400"
  })), "Sign Up")))));
}

function Reset1(_ref) {
  var title = _ref.title,
      description = _ref.description,
      logo = _ref.logo,
      loginUrl = _ref.loginUrl,
      onReset = _ref.onReset,
      action = _ref.action,
      error = _ref.error;

  var _useState = React.useState(null),
      email = _useState[0],
      setEmail = _useState[1];

  var _useState2 = React.useState(error),
      errorMsg = _useState2[0],
      setErrorMsg = _useState2[1];

  var _loginUrl = loginUrl;

  var _title = title || "Reset Account Password";

  var _description = description || "By resetting your password you can get access to all the awesome features our system has to offer.";

  var _logo = logo || /*#__PURE__*/React__default.createElement(outline.CubeIcon, {
    className: "p-2 mx-auto w-16 h-16 text-white bg-blue-500 rounded-full"
  });

  React.useEffect(function () {
    setErrorMsg(error);
  }, [error]);

  var isValid = function isValid() {
    setErrorMsg(null);

    if (email) {
      if (email.includes("@") && email.includes(".")) return true;
      setErrorMsg("Email address must be in a valid format.");
      return false;
    }

    setErrorMsg("Email is required.");
    return false;
  };

  return /*#__PURE__*/React__default.createElement("div", {
    className: "min-h-screen flex items-center justify-center bg-gray-50 py-12 px-4 sm:px-6 lg:px-8"
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "max-w-md w-full space-y-8"
  }, /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement("div", {
    className: "mx-auto"
  }, _logo), /*#__PURE__*/React__default.createElement("h2", {
    className: "mt-6 text-center text-3xl font-medium text-gray-900"
  }, _title), /*#__PURE__*/React__default.createElement("h4", {
    className: "mt-6 text-center text-sm text-gray-500"
  }, _description)), /*#__PURE__*/React__default.createElement("form", {
    className: "mt-8 space-y-6",
    method: "POST",
    action: action
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "rounded-md shadow-sm -space-y-px"
  }, /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement("label", {
    htmlFor: "email-address",
    className: "sr-only"
  }, "Email address"), /*#__PURE__*/React__default.createElement("input", {
    id: "username",
    name: "username",
    type: "email",
    autoComplete: "email",
    required: true,
    className: "appearance-none rounded-none relative block w-full px-3 py-2 border  border-gray-300 placeholder-gray-500 text-gray-900 rounded-t-md focus:outline-none focus:ring-blue-500 focus:border-blue-500 focus:z-10 sm:text-sm",
    placeholder: "Email address",
    onChange: function onChange(e) {
      setEmail(e.target.value);
    }
  }))), /*#__PURE__*/React__default.createElement("div", {
    className: "flex items-center justify-center"
  }, _loginUrl && /*#__PURE__*/React__default.createElement("div", {
    className: "text-sm"
  }, /*#__PURE__*/React__default.createElement("a", {
    href: _loginUrl,
    className: "font-medium text-blue-600 hover:text-blue-500"
  }, "Sign in instead?"))), errorMsg && /*#__PURE__*/React__default.createElement("div", {
    className: "text-red-500"
  }, errorMsg), /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement("button", {
    type: action ? "submit" : "button",
    className: "group relative w-full flex justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-md text-white bg-blue-600  hover:bg-blue-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-blue-500",
    onClick: function onClick() {
      if (onReset && isValid()) onReset({
        username: email
      });
    }
  }, /*#__PURE__*/React__default.createElement("span", {
    className: "absolute left-0 inset-y-0 flex items-center pl-3"
  }, /*#__PURE__*/React__default.createElement(outline.LockClosedIcon, {
    className: "h-5 w-5 text-blue-500 group-hover:text-blue-400"
  })), "Reset")))));
}

function Statistic3(_ref) {
  var title = _ref.title,
      description = _ref.description,
      stats = _ref.stats,
      className = _ref.className;

  var _title = title || "Master Cleanse Reliac Heirloom";

  var _description = description || "Whatever cardigan tote bag tumblr hexagon brooklyn asymmetrical gentrify, subway tile poke farm-to-table. Franzen you probably haven't heard of them man bun deep jianbing selfies heirloom prism food truck ugh squid celiac humblebrag.";

  var _stats = stats || [{
    title: "Downloads",
    value: "2.7K",
    icon: /*#__PURE__*/React__default.createElement(outline.CloudDownloadIcon, {
      className: "text-blue-500 w-12 h-12 mb-3 inline-block"
    })
  }, {
    title: "Users",
    value: "1.3K",
    icon: /*#__PURE__*/React__default.createElement(outline.UserGroupIcon, {
      className: "text-blue-500 w-12 h-12 mb-3 inline-block"
    })
  }, {
    title: "Files",
    value: "74",
    icon: /*#__PURE__*/React__default.createElement(outline.DocumentIcon, {
      className: "text-blue-500 w-12 h-12 mb-3 inline-block"
    })
  }, {
    title: "Locations",
    value: "46",
    icon: /*#__PURE__*/React__default.createElement(outline.LocationMarkerIcon, {
      className: "text-blue-500 w-12 h-12 mb-3 inline-block"
    })
  }];

  return /*#__PURE__*/React__default.createElement("section", {
    className: "text-gray-600 body-font " + (className ? className : "")
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "container px-5 py-24 mx-auto"
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "flex flex-col text-center w-full mb-20"
  }, typeof _title === "string" ? /*#__PURE__*/React__default.createElement("h1", {
    className: "sm:text-3xl text-2xl font-medium title-font mb-4 text-gray-900"
  }, _title) : _title, typeof _description === "string" ? /*#__PURE__*/React__default.createElement("p", {
    className: "lg:w-2/3 mx-auto leading-relaxed text-base"
  }, _description) : _description), /*#__PURE__*/React__default.createElement("div", {
    className: "flex flex-wrap -m-4 text-center"
  }, _stats.map(function (s, i) {
    return /*#__PURE__*/React__default.createElement("div", {
      className: "p-4 md:w-1/4 sm:w-1/2 w-full",
      key: i
    }, /*#__PURE__*/React__default.createElement("div", {
      className: "border-2 border-gray-200 px-4 py-6 rounded-lg"
    }, s.icon, typeof s.value !== "string" ? s.value : /*#__PURE__*/React__default.createElement("h2", {
      className: "title-font font-medium text-3xl text-gray-900"
    }, s.value), typeof s.title === "string" ? /*#__PURE__*/React__default.createElement("p", {
      className: "leading-relaxed"
    }, s.title) : s.title));
  }))));
}

function Steps3(_ref) {
  var title = _ref.title,
      steps = _ref.steps,
      className = _ref.className;

  var _title = title || /*#__PURE__*/React__default.createElement("h1", {
    className: "pt-24 sm:text-3xl text-2xl font-medium title-font text-center text-gray-900 mb-20"
  }, "As Simple as One, Two, Three, Four");

  var _steps = steps || [{
    title: "Shooting Stars",
    description: "VHS cornhole pop-up, try-hard 8-bit iceland helvetica. Kinfolk bespoke try-hard cliche palo santo offal.",
    step: "1",
    icon: /*#__PURE__*/React__default.createElement(outline.ShieldCheckIcon, {
      className: "w-12 h-12"
    })
  }, {
    title: "The Catalyzer",
    description: "VHS cornhole pop-up, try-hard 8-bit iceland helvetica. Kinfolk bespoke try-hard cliche palo santo offal.",
    step: "2",
    icon: /*#__PURE__*/React__default.createElement(outline.TrendingUpIcon, {
      className: "w-12 h-12"
    })
  }, {
    title: "The 400 Blows",
    description: "VHS cornhole pop-up, try-hard 8-bit iceland helvetica. Kinfolk bespoke try-hard cliche palo santo offal.",
    step: "3",
    icon: /*#__PURE__*/React__default.createElement(outline.CloudDownloadIcon, {
      className: "w-12 h-12"
    })
  }, {
    title: "Neptune",
    description: "VHS cornhole pop-up, try-hard 8-bit iceland helvetica. Kinfolk bespoke try-hard cliche palo santo offal.",
    step: "4",
    icon: /*#__PURE__*/React__default.createElement(outline.UserIcon, {
      className: "w-12 h-12"
    })
  }];

  return /*#__PURE__*/React__default.createElement("section", {
    className: "text-gray-600 body-font " + (className ? className : "")
  }, typeof _title === "string" ? /*#__PURE__*/React__default.createElement("h1", {
    className: "sm:text-3xl text-2xl font-medium title-font text-center text-gray-900 mb-20"
  }, _title) : _title, /*#__PURE__*/React__default.createElement("div", {
    className: "container px-5 pb-24 mx-auto flex flex-wrap"
  }, _steps.map(function (s, i) {
    return /*#__PURE__*/React__default.createElement("div", {
      key: i,
      className: "flex relative pt-10 pb-20 sm:items-center md:w-2/3 mx-auto"
    }, /*#__PURE__*/React__default.createElement("div", {
      className: "h-full w-6 absolute inset-0 flex items-center justify-center"
    }, /*#__PURE__*/React__default.createElement("div", {
      className: "h-full w-1 bg-gray-200 pointer-events-none"
    })), /*#__PURE__*/React__default.createElement("div", {
      className: "flex-shrink-0 w-6 h-6 rounded-full mt-10 sm:mt-0 inline-flex items-center justify-center bg-indigo-500 text-white relative z-10 title-font font-medium text-sm"
    }, s.step), /*#__PURE__*/React__default.createElement("div", {
      className: "flex-grow md:pl-8 pl-6 flex sm:items-center items-start flex-col sm:flex-row"
    }, /*#__PURE__*/React__default.createElement("div", {
      className: "flex-shrink-0 w-24 h-24 bg-indigo-100 text-indigo-500 rounded-full inline-flex items-center justify-center"
    }, s.icon), /*#__PURE__*/React__default.createElement("div", {
      className: "flex-grow sm:pl-6 mt-6 sm:mt-0"
    }, typeof s.title === "string" ? /*#__PURE__*/React__default.createElement("h2", {
      className: "font-medium title-font text-gray-900 mb-1 text-xl"
    }, s.title) : s.title, typeof s.description === "string" ? /*#__PURE__*/React__default.createElement("p", {
      className: "leading-relaxed"
    }, s.description) : s.description)));
  })));
}

function Tabs1(_ref) {
  var tabs = _ref.tabs,
      className = _ref.className;

  var _useState = React.useState(0),
      selected = _useState[0],
      setSelected = _useState[1];

  var _tabs = tabs || [{
    title: "Description",
    content: /*#__PURE__*/React__default.createElement("p", null, "Description locavore kickstarter distillery. Mixtape chillwave tumeric sriracha taximy chia microdosing tilde DIY. XOXO fam inxigo juiceramps cornhole raw denim forage brooklyn. Everyday carry seitan poutine tumeric. Gastropub blue bottle austin listicle pour-over, neutra jean.")
  }, {
    title: "Reviews",
    content: /*#__PURE__*/React__default.createElement("p", null, "Reviews locavore kickstarter distillery. Mixtape chillwave tumeric sriracha taximy chia microdosing tilde DIY. XOXO fam inxigo juiceramps cornhole raw denim forage brooklyn. Everyday carry seitan poutine tumeric. Gastropub blue bottle austin listicle pour-over, neutra jean.")
  }, {
    title: "Details",
    content: /*#__PURE__*/React__default.createElement("p", null, "Details locavore kickstarter distillery. Mixtape chillwave tumeric sriracha taximy chia microdosing tilde DIY. XOXO fam inxigo juiceramps cornhole raw denim forage brooklyn. Everyday carry seitan poutine tumeric. Gastropub blue bottle austin listicle pour-over, neutra jean.")
  }];

  return /*#__PURE__*/React__default.createElement("div", {
    className: className
  }, /*#__PURE__*/React__default.createElement("div", {
    className: "flex mb-4"
  }, _tabs.map(function (t, i) {
    var cls = selected == i ? "text-blue-500 border-blue-500" : "text-gray-900 border-gray-300";
    return /*#__PURE__*/React__default.createElement("a", {
      key: i,
      className: "flex-grow border-b-2 py-2 text-lg px-1 cursor-pointer " + cls,
      onClick: function onClick() {
        return setSelected(i);
      }
    }, t.title);
  })), /*#__PURE__*/React__default.createElement("div", null, _tabs[selected].content));
}

exports.Blog1 = Blog1;
exports.Card1 = Card1;
exports.Content3 = Content3;
exports.Dashboard1 = Dashboard1;
exports.Ecommerce1 = Ecommerce1;
exports.Ecommerce2 = Ecommerce2;
exports.Email1 = Email1;
exports.Feature1 = Feature1;
exports.Footer1 = Footer1;
exports.Footer4 = Footer4;
exports.Form = Form;
exports.Grid1 = Grid1;
exports.Header2 = Header2;
exports.Hero1 = Hero1;
exports.Input = Input;
exports.Layout1 = Form1;
exports.Login1 = Login1;
exports.Logo1 = Logo1;
exports.Modal1 = Modal1;
exports.Pricing1 = Pricing1;
exports.Register1 = Register1;
exports.Reset1 = Reset1;
exports.Select = Select;
exports.Statistics3 = Statistic3;
exports.Steps3 = Steps3;
exports.Table1 = Table1;
exports.Tabs1 = Tabs1;
exports.TextArea = TextArea;
